diff --git a/pcmanfm/settings.cpp b/pcmanfm/settings.cpp
index b19ee97..660c740 100644
--- a/pcmanfm/settings.cpp
+++ b/pcmanfm/settings.cpp
@@ -147,7 +147,7 @@ bool Settings::loadFile(QString filePath) {
   if(fallbackIconThemeName_.isEmpty()) {
     // FIXME: we should choose one from installed icon themes or get
     // the value from XSETTINGS instead of hard code a fallback value.
-    fallbackIconThemeName_ = "elementary"; // fallback icon theme name
+    fallbackIconThemeName_ = "oxygen"; // fallback icon theme name
   }
   suCommand_ = settings.value("SuCommand", "gksu %s").toString();
   setTerminal(settings.value("Terminal", "xterm").toString());
@@ -192,10 +192,10 @@ bool Settings::loadFile(QString filePath) {
   settings.endGroup();
 
   settings.beginGroup("Volume");
-  mountOnStartup_ = settings.value("MountOnStartup", true).toBool();
-  mountRemovable_ = settings.value("MountRemovable", true).toBool();
+  mountOnStartup_ = settings.value("MountOnStartup", false).toBool();
+  mountRemovable_ = settings.value("MountRemovable", false).toBool();
   autoRun_ = settings.value("AutoRun", true).toBool();
-  closeOnUnmount_ = settings.value("CloseOnUnmount", true).toBool();
+  closeOnUnmount_ = settings.value("CloseOnUnmount", false).toBool();
   settings.endGroup();
 
   settings.beginGroup("Thumbnail");
