diff --git a/src/ippool.c b/src/ippool.c
index cea1dcc..ece7ba7 100644
--- a/src/ippool.c
+++ b/src/ippool.c
@@ -232,12 +232,13 @@ static struct address_info *lookup_info(int index, uint32_t start)
 
 static bool is_private_address(uint32_t address)
 {
-	unsigned int a, b;
+	unsigned int a, b, c;
 
 	a = (address & 0xff000000) >> 24;
 	b = (address & 0x00ff0000) >> 16;
+	c = (address & 0x0000ff00) >> 8;
 
-	if (a == 10 || (a == 192 && b == 168) ||
+	if (a == 10 || (a == 192 && b == 168 && c >= 7) ||
 					(a == 172 && (b >= 16 && b <= 31)))
 		return true;
 
diff --git a/src/tethering.c b/src/tethering.c
index 3153349..86e78dc 100644
--- a/src/tethering.c
+++ b/src/tethering.c
@@ -485,7 +485,7 @@ int __connman_private_network_request(DBusMessage *msg, const char *owner)
 	pn->fd = fd;
 	pn->interface = iface;
 	pn->index = index;
-	pn->pool = __connman_ippool_create(pn->index, 1, 1, ippool_disconnect, pn);
+	pn->pool = __connman_ippool_create(pn->index, 2, 1, ippool_disconnect, pn);
 	if (!pn->pool) {
 		errno = -ENOMEM;
 		goto error;
