Index: b/interface/isl.py
===================================================================
--- a/interface/isl.py
+++ b/interface/isl.py
@@ -1,6 +1,6 @@
 from ctypes import *
 
-isl = cdll.LoadLibrary("libisl.so")
+isl = cdll.LoadLibrary("libisl.so.10")
 libc = cdll.LoadLibrary("libc.so.6")
 
 class Error(Exception):
Index: b/interface/isl.py.top
===================================================================
--- a/interface/isl.py.top
+++ b/interface/isl.py.top
@@ -1,6 +1,6 @@
 from ctypes import *
 
-isl = cdll.LoadLibrary("libisl.so")
+isl = cdll.LoadLibrary("libisl.so.10")
 libc = cdll.LoadLibrary("libc.so.6")
 
 class Error(Exception):
