From: Guillaume Grossetie <ggrossetie@gmail.com>
Subject: Load data from /usr/share/asciidoctor path
Bug-Debian: https://bugs.debian.org/788052
--- a/lib/asciidoctor.rb
+++ b/lib/asciidoctor.rb
@@ -178,7 +178,7 @@ module Asciidoctor
   LIB_PATH = ::File.join ROOT_PATH, 'lib'
 
   # The absolute data path of the Asciidoctor RubyGem
-  DATA_PATH = ::File.join ROOT_PATH, 'data'
+  DATA_PATH = ::File.exist?(asciidoctor_lib_path = ::File.join(ROOT_PATH, 'data')) ? asciidoctor_lib_path : ::File.join(ROOT_PATH, '../../share/asciidoctor')
 
   # The user's home directory, as best we can determine it
   # NOTE not using infix rescue for performance reasons, see: https://github.com/jruby/jruby/issues/1816
