From: Guillaume Grossetie <ggrossetie@gmail.com>
Subject: Exclude lib/ directory from tests
--- a/Rakefile
+++ b/Rakefile
@@ -20,7 +20,7 @@ begin
   Rake::TestTask.new(:test) do |test|
     prepare_test_env
     puts %(LANG: #{ENV['LANG']}) if ENV.key? 'TRAVIS_BUILD_ID'
-    test.libs << 'test'
+    test.libs = ['test']
     test.pattern = 'test/**/*_test.rb'
     test.verbose = true
     test.warning = true
--- a/test/test_helper.rb
+++ b/test/test_helper.rb
@@ -8,7 +8,7 @@ end
 
 require 'simplecov' if ENV['COVERAGE'] == 'true'
 
-require File.join(ASCIIDOCTOR_PROJECT_DIR, 'lib', 'asciidoctor')
+require 'asciidoctor'

 require 'socket'
 require 'nokogiri'
