Index: asciidoctor/test/reader_test.rb
===================================================================
--- asciidoctor.orig/test/reader_test.rb	2016-02-29 17:15:14.900028599 +0100
+++ asciidoctor/test/reader_test.rb	2016-02-29 18:16:05.506299075 +0100
@@ -618,6 +618,7 @@
       end
 
       test 'unreadable file referenced by include directive is replaced by warning' do
+        skip 'file is readable in Debian build'
         include_file = File.join DIRNAME, 'fixtures', 'chapter-a.adoc'
         FileUtils.chmod 0000, include_file
         input = <<-EOS
