From 61d5e486562b54e6aa30e98d4587d1be31c285f1 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 3 Jan 2019 09:53:37 -0600
Subject: [PATCH] fix: opencl-firmware git repo

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makefile b/makefile
index 7cab2e7..2e33960 100644
--- a/makefile
+++ b/makefile
@@ -136,7 +136,7 @@ eve_submodule:
 ifeq ($(USE_EXTERNAL_FW_REPO),1)
 FW_REPO=git://git.ti.com/opencl/opencl-firmware.git
 else
-FW_REPO=ssh://git@bitbucket.itg.ti.com/mctools/opencl-firmware.git
+FW_REPO=git://git.ti.com/opencl/opencl-firmware.git
 endif
 
 opencl-firmware:
-- 
2.20.1

