# DP: fixes an uninitialized memory issue in adlc
--- jdk8/hotspot/src/share/vm/adlc/formsopt.cpp	2014-07-03 21:56:12.000000000 +0200
+++ jdk8/hotspot/src/share/vm/adlc/formsopt.cpp	2014-07-14 11:43:21.900408570 +0200
@@ -347,6 +347,7 @@
   _return_value = NULL;
   _c_return_value = NULL;
   _interpreter_frame_pointer_reg = NULL;
+  _cisc_spilling_operand_name = NULL;
 }
 
 FrameForm::~FrameForm() {
