Index: b/jdk/src/solaris/classes/sun/awt/X11/XDataTransferer.java
===================================================================
--- a/jdk/src/solaris/classes/sun/awt/X11/XDataTransferer.java
+++ b/jdk/src/solaris/classes/sun/awt/X11/XDataTransferer.java
@@ -281,14 +281,14 @@ public class XDataTransferer extends Dat
             reader = new BufferedReader(new InputStreamReader(stream, charset));
             String line;
             ArrayList<URI> uriList = new ArrayList<URI>();
-            URI uri;
             while ((line = reader.readLine()) != null) {
                 try {
-                    uri = new URI(line);
+                    if (!"\0".equals(line)) {
+                        uriList.add(new URI(line));
+                    }
                 } catch (URISyntaxException uriSyntaxException) {
                     throw new IOException(uriSyntaxException);
                 }
-                uriList.add(uri);
             }
             return uriList.toArray(new URI[uriList.size()]);
         } finally {
