Index: b/common/autoconf/toolchain.m4
===================================================================
--- a/common/autoconf/toolchain.m4
+++ b/common/autoconf/toolchain.m4
@@ -621,7 +621,9 @@ AC_DEFUN_ONCE([TOOLCHAIN_SETUP_COMPILER_
     LDEXE="$CC"
     LDCXX="$CXX"
     LDEXECXX="$CXX"
-    POST_STRIP_CMD="$STRIP -g"
+    if test -z "$POST_STRIP_CMD" ; then
+      POST_STRIP_CMD="$STRIP -g"
+    fi
 
     # Linking is different on MacOSX
     if test "x$OPENJDK_TARGET_OS" = xmacosx; then
