--- a/chrome/browser/ui/views/frame/browser_frame.cc
+++ b/chrome/browser/ui/views/frame/browser_frame.cc
@@ -30,6 +30,7 @@
 #include "ui/gfx/font_list.h"
 #include "ui/views/controls/menu/menu_runner.h"
 #include "ui/views/widget/native_widget.h"
+#include "ui/views/widget/native_widget_aura.h"
 
 #if defined(OS_CHROMEOS)
 #include "ash/session/session_state_delegate.h"
@@ -137,6 +138,8 @@ BrowserNonClientFrameView* BrowserFrame:
   return browser_frame_view_;
 }
 
+void BrowserFrame::SendSessionIdToWindow() { }
+
 bool BrowserFrame::UseCustomFrame() const {
   return native_browser_frame_->UseCustomFrame();
 }
--- a/chrome/browser/ui/views/frame/browser_frame.h
+++ b/chrome/browser/ui/views/frame/browser_frame.h
@@ -95,6 +95,8 @@ class BrowserFrame
   // Called when BrowserView creates all it's child views.
   void OnBrowserViewInitViewsComplete();
 
+  void SendSessionIdToWindow();
+
   // Overridden from views::Widget:
   views::internal::RootView* CreateRootView() override;
   views::NonClientFrameView* CreateNonClientFrameView() override;
--- a/chrome/browser/ui/views/frame/browser_view.cc
+++ b/chrome/browser/ui/views/frame/browser_view.cc
@@ -712,6 +712,7 @@ void BrowserView::Show() {
   force_location_bar_focus_ = false;
 
   browser()->OnWindowDidShow();
+  frame_->SendSessionIdToWindow();
 
   chrome::MaybeShowInvertBubbleView(this);
 }
--- a/ui/aura/client/aura_constants.cc
+++ b/ui/aura/client/aura_constants.cc
@@ -8,6 +8,7 @@
 #include "ui/gfx/geometry/rect.h"
 
 DECLARE_EXPORTED_WINDOW_PROPERTY_TYPE(AURA_EXPORT, bool)
+DECLARE_EXPORTED_WINDOW_PROPERTY_TYPE(AURA_EXPORT, uint32_t)
 DECLARE_EXPORTED_WINDOW_PROPERTY_TYPE(AURA_EXPORT, ui::ModalType)
 DECLARE_EXPORTED_WINDOW_PROPERTY_TYPE(AURA_EXPORT, gfx::Rect*)
 DECLARE_EXPORTED_WINDOW_PROPERTY_TYPE(AURA_EXPORT, ui::InputMethod*)
@@ -34,6 +35,7 @@ DEFINE_WINDOW_PROPERTY_KEY(ui::ModalType
 DEFINE_OWNED_WINDOW_PROPERTY_KEY(gfx::Rect, kRestoreBoundsKey, NULL);
 DEFINE_WINDOW_PROPERTY_KEY(
     ui::WindowShowState, kRestoreShowStateKey, ui::SHOW_STATE_DEFAULT);
+DEFINE_WINDOW_PROPERTY_KEY(uint32_t, kSessionIdKey, 0);
 DEFINE_WINDOW_PROPERTY_KEY(
     ui::WindowShowState, kShowStateKey, ui::SHOW_STATE_DEFAULT);
 DEFINE_WINDOW_PROPERTY_KEY(int, kTopViewInset, 0);
--- a/ui/aura/client/aura_constants.h
+++ b/ui/aura/client/aura_constants.h
@@ -67,6 +67,9 @@ AURA_EXPORT extern const WindowProperty<
 // the web contents for app windows and varies for fullscreen windows.
 AURA_EXPORT extern const aura::WindowProperty<int>* const kTopViewInset;
 
+AURA_EXPORT extern const WindowProperty<uint32_t>* const kSessionIdKey;
+
+
 // Alphabetical sort.
 
 }  // namespace client
--- a/ui/aura/window_tree_host_x11.h
+++ b/ui/aura/window_tree_host_x11.h
@@ -38,6 +38,8 @@ class AURA_EXPORT WindowTreeHostX11 : pu
   bool CanDispatchEvent(const ui::PlatformEvent& event) override;
   uint32_t DispatchEvent(const ui::PlatformEvent& event) override;
 
+//  void SetSessionID(uint32 s) OVERRIDE;
+
   // WindowTreeHost:
   ui::EventSource* GetEventSource() override;
   gfx::AcceleratedWidget GetAcceleratedWidget() override;
--- a/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc
+++ b/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc
@@ -591,6 +591,11 @@ void DesktopNativeWidgetAura::ViewRemove
   drop_helper_->ResetTargetViewIfEquals(view);
 }
 
+void DesktopNativeWidgetAura::SetSessionID(unsigned int sessionid) {
+  desktop_window_tree_host_->SetSessionID(sessionid);
+  VLOG(1) << "session id Sent to treehost";
+}
+
 void DesktopNativeWidgetAura::SetNativeWindowProperty(const char* name,
                                                       void* value) {
   if (content_window_)
--- a/ui/views/widget/desktop_aura/desktop_native_widget_aura.h
+++ b/ui/views/widget/desktop_aura/desktop_native_widget_aura.h
@@ -89,6 +89,8 @@ class VIEWS_EXPORT DesktopNativeWidgetAu
   // we are being activated/deactivated.
   void HandleActivationChanged(bool active);
 
+  void SetSessionID(uint32_t sessionid);
+
  protected:
   // Overridden from internal::NativeWidgetPrivate:
   void InitNativeWidget(const Widget::InitParams& params) override;
--- a/ui/views/widget/desktop_aura/desktop_window_tree_host.h
+++ b/ui/views/widget/desktop_aura/desktop_window_tree_host.h
@@ -81,6 +81,8 @@ class VIEWS_EXPORT DesktopWindowTreeHost
 
   virtual bool IsVisible() const = 0;
 
+  virtual void SetSessionID(uint32_t sessionid) = 0;
+
   virtual void SetSize(const gfx::Size& size) = 0;
   virtual void StackAbove(aura::Window* window) = 0;
   virtual void StackAtTop() = 0;
--- a/ui/views/widget/desktop_aura/desktop_window_tree_host_x11.cc
+++ b/ui/views/widget/desktop_aura/desktop_window_tree_host_x11.cc
@@ -9,6 +9,7 @@
 #include <X11/Xutil.h>
 #include <X11/extensions/XInput2.h>
 #include <X11/extensions/shape.h>
+#include <signal.h>
 
 #include <utility>
 
@@ -111,6 +112,7 @@ const char* kAtomsToCache[] = {
   "_NET_WM_WINDOW_TYPE_NORMAL",
   "_NET_WM_WINDOW_TYPE_NOTIFICATION",
   "_NET_WM_WINDOW_TYPE_TOOLTIP",
+  "CHROMIUM_LINUX_SESSION_ID_PROPERTY",
   "XdndActionAsk",
   "XdndActionCopy",
   "XdndActionLink",
@@ -1432,6 +1434,22 @@ void DesktopWindowTreeHostX11::UpdateMin
   XSetWMNormalHints(xdisplay_, xwindow_, &hints);
 }
 
+void DesktopWindowTreeHostX11::SetSessionID(uint32_t session_id) {
+  if (session_id != 0) {
+    XChangeProperty(xdisplay_,
+                    xwindow_,
+                    atom_cache_.GetAtom("CHROMIUM_LINUX_SESSION_ID_PROPERTY"),
+                    XA_CARDINAL,
+                    32,
+                    PropModeReplace,
+                    reinterpret_cast<unsigned char*>(&session_id),
+                    1);
+    VLOG(1) << "Sent session id " << session_id << " to X window property";
+  } else {
+    VLOG(1) << "Session id is unlikely number.";
+  }
+}
+
 void DesktopWindowTreeHostX11::UpdateWMUserTime(
     const ui::PlatformEvent& event) {
   if (!IsActive())
--- a/ui/views/widget/desktop_aura/desktop_window_tree_host_x11.h
+++ b/ui/views/widget/desktop_aura/desktop_window_tree_host_x11.h
@@ -86,6 +86,8 @@ class VIEWS_EXPORT DesktopWindowTreeHost
   // internal list of open windows.
   static void CleanUpWindowList(void (*func)(aura::Window* window));
 
+  void SetSessionID(uint32_t sessionid);
+
  protected:
   // Overridden from DesktopWindowTreeHost:
   void Init(aura::Window* content_window,
--- a/ui/views/widget/native_widget.h
+++ b/ui/views/widget/native_widget.h
@@ -26,6 +26,8 @@ class VIEWS_EXPORT NativeWidget {
  public:
   virtual ~NativeWidget() {}
 
+  virtual void SetSessionID(uint32_t sessionid) = 0;
+
  private:
   friend class Widget;
 
--- a/ui/views/widget/native_widget_aura.cc
+++ b/ui/views/widget/native_widget_aura.cc
@@ -62,6 +62,8 @@
 #include "ui/views/widget/desktop_aura/desktop_window_tree_host.h"
 #endif
 
+#include <signal.h>
+
 namespace views {
 
 namespace {
@@ -332,6 +334,11 @@ void NativeWidgetAura::CenterWindow(cons
   window_->SetBounds(window_bounds);
 }
 
+void NativeWidgetAura::SetSessionID(uint32_t sessionid) {
+  VLOG(1) << "I have no idea.";
+  raise(SIGABRT);
+}
+
 void NativeWidgetAura::GetWindowPlacement(
     gfx::Rect* bounds,
     ui::WindowShowState* show_state) const {
--- a/ui/views/widget/native_widget_aura.h
+++ b/ui/views/widget/native_widget_aura.h
@@ -50,6 +50,8 @@ class VIEWS_EXPORT NativeWidgetAura
       internal::NativeWidgetPrivate* native_widget,
       aura::Window* window);
 
+  virtual void SetSessionID(uint32_t sessionid) override;
+
   // Overridden from internal::NativeWidgetPrivate:
   void InitNativeWidget(const Widget::InitParams& params) override;
   void OnWidgetInitDone() override;
