Description: Fix gn bootstrap build
Origin: https://chromium.googlesource.com/chromium/src/+/379e35f6f3eaa41a97f2659249509ca599749b27
Author: Olivier Tilloy <olivier.tilloy@canonical.com>

--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -384,6 +384,7 @@ def write_gn_ninja(path, root_gen_dir, o
       'base/base_paths.cc',
       'base/base_switches.cc',
       'base/build_time.cc',
+      'base/callback_helpers.cc',
       'base/callback_internal.cc',
       'base/command_line.cc',
       'base/debug/activity_tracker.cc',
