--- a/chrome/common/chrome_content_client.cc
+++ b/chrome/common/chrome_content_client.cc
@@ -204,9 +204,8 @@ void ComputeBuiltInPlugins(std::vector<c
     widevine_cdm.path = adapter_path;
     widevine_cdm.name = kWidevineCdmDisplayName;
     widevine_cdm.description =
-        base::StringPrintf("%s (version: " WIDEVINE_CDM_VERSION_STRING ")",
+	    (
                            kWidevineCdmDescription);
-    widevine_cdm.version = WIDEVINE_CDM_VERSION_STRING;
     content::WebPluginMimeType widevine_cdm_mime_type(
         kWidevineCdmPluginMimeType, kWidevineCdmPluginExtension,
         kWidevineCdmPluginMimeTypeDescription);
