From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Tue, 2 Jun 2015 17:44:31 +0200
Subject: Add Debian specific SOVERSION

---
 nodelet/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/nodelet/CMakeLists.txt b/nodelet/CMakeLists.txt
index 5d9eea4..2773adf 100644
--- a/nodelet/CMakeLists.txt
+++ b/nodelet/CMakeLists.txt
@@ -31,6 +31,7 @@ include_directories(include ${catkin_INCLUDE_DIRS} ${BOOST_INCLUDE_DIRS})
 add_library(nodeletlib src/nodelet_class.cpp src/loader.cpp src/callback_queue.cpp src/callback_queue_manager.cpp)
 target_link_libraries(nodeletlib ${catkin_LIBRARIES} ${BOOST_LIBRARIES})
 add_dependencies(nodeletlib ${nodelet_EXPORTED_TARGETS})
+set_target_properties(nodeletlib PROPERTIES VERSION ${nodelet_VERSION} SOVERSION "0d")
 
 add_executable(nodelet src/nodelet.cpp)
 target_link_libraries(nodelet nodeletlib ${UUID_LIBRARIES} ${catkin_LIBRARIES} ${BOOST_LIBRARIES})
