From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Fri, 24 Apr 2015 13:22:39 +0200
Subject: Disable geneus for now

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a6700e6..7052f27 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,4 +3,4 @@ project(message_generation)
 
 find_package(catkin REQUIRED)
 
-catkin_package(CATKIN_DEPENDS gencpp geneus genlisp genmsg genpy)
+catkin_package(CATKIN_DEPENDS gencpp genlisp genmsg genpy)
