From: Svante Signel <svante.signell@gmail.com>
Date: Tue, 6 Oct 2015 16:35:34 +0100
Subject: Add GNU/Hurd support

---
 configure.ac | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/configure.ac b/configure.ac
index 2ea486a..309f2b5 100755
--- a/configure.ac
+++ b/configure.ac
@@ -130,6 +130,7 @@ LIBS="${PREVIOUS_LIBS}"
 czmq_on_mingw32="no"
 czmq_on_cygwin="no"
 czmq_on_android="no"
+czmq_on_gnu="no"
 
 # Host specific checks
 AC_CANONICAL_HOST
@@ -222,6 +223,11 @@ case "${host_os}" in
         AC_DEFINE(CZMQ_HAVE_CYGWIN, 1, [Have Cygwin])
         czmq_on_cygwin="yes"
         ;;
+    gnu*)
+        # Define on GNU/Hurd to enable all library features
+        CPPFLAGS="-D_GNU_SOURCE $CPPFLAGS"
+        AC_DEFINE(CZMQ_HAVE_GNU, 1, [Have GNU/Hurd OS])
+        ;;
     *)
         AC_MSG_ERROR([unsupported system: ${host_os}.])
         ;;
