From: Youhei SASAKI <uwabami@gfd-dennou.org>
Date: Tue, 3 Nov 2015 10:07:12 +0900
Subject: Drop Ruby Minitest Profile

Signed-off-by: Youhei SASAKI <uwabami@gfd-dennou.org>
---
 Gemfile        | 1 -
 test/helper.rb | 1 -
 2 files changed, 2 deletions(-)

diff --git a/Gemfile b/Gemfile
index 9b64809..9ddb15b 100644
--- a/Gemfile
+++ b/Gemfile
@@ -17,7 +17,6 @@ group :test do
   gem 'jekyll_test_plugin'
   gem 'jekyll_test_plugin_malicious'
   gem 'minitest-reporters'
-  gem 'minitest-profile'
   gem 'minitest'
   gem 'rspec-mocks'
 
diff --git a/test/helper.rb b/test/helper.rb
index dc02176..25dfc2e 100644
--- a/test/helper.rb
+++ b/test/helper.rb
@@ -15,7 +15,6 @@ require 'rubygems'
 require 'ostruct'
 require 'minitest/autorun'
 require 'minitest/reporters'
-require 'minitest/profile'
 require 'rspec/mocks'
 
 require 'jekyll'
