From: Youhei SASAKI <uwabami@gfd-dennou.org>
Date: Tue, 3 Nov 2015 10:10:47 +0900
Subject: Downgrading RDoc as 4.1

Signed-off-by: Youhei SASAKI <uwabami@gfd-dennou.org>
---
 Gemfile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Gemfile b/Gemfile
index 9ddb15b..19936a2 100644
--- a/Gemfile
+++ b/Gemfile
@@ -3,9 +3,9 @@ gemspec name: 'jekyll'
 
 gem 'rake', '~> 10.1'
 group :development do
-  gem 'rdoc', '~> 4.2'
+  gem 'rdoc', '~> 4.1'
   gem 'launchy', '~> 2.3'
-  gem 'toml', '~> 0.1.0'
+  gem 'toml', '=> 0.1.0'
   gem 'pry'
 end
 
