From: Youhei SASAKI <uwabami@gfd-dennou.org>
Date: Tue, 3 Nov 2015 10:20:17 +0900
Subject: Remove Launchy Dependencies

Signed-off-by: Youhei SASAKI <uwabami@gfd-dennou.org>
---
 Gemfile  | 1 -
 Rakefile | 3 +--
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/Gemfile b/Gemfile
index 19936a2..eb1f9cb 100644
--- a/Gemfile
+++ b/Gemfile
@@ -4,7 +4,6 @@ gemspec name: 'jekyll'
 gem 'rake', '~> 10.1'
 group :development do
   gem 'rdoc', '~> 4.1'
-  gem 'launchy', '~> 2.3'
   gem 'toml', '=> 0.1.0'
   gem 'pry'
 end
diff --git a/Rakefile b/Rakefile
index ff473e6..937fd26 100644
--- a/Rakefile
+++ b/Rakefile
@@ -134,7 +134,6 @@ end
 namespace :site do
   desc "Generate and view the site locally"
   task :preview do
-    require "launchy"
     require "jekyll"
 
     # Yep, it's a hack! Wait a few seconds for the Jekyll site to generate and
@@ -142,7 +141,7 @@ namespace :site do
     Thread.new do
       sleep 4
       puts "Opening in browser..."
-      Launchy.open("http://localhost:4000")
+      sh "xdg-open http://localhost:4000"
     end
 
     # Generate the site in server mode.
