From: Youhei SASAKI <uwabami@gfd-dennou.org>
Date: Wed, 4 Nov 2015 15:01:18 +0900
Subject: Fix Gemfile versioning: toml

Signed-off-by: Youhei SASAKI <uwabami@gfd-dennou.org>
---
 Gemfile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Gemfile b/Gemfile
index eb1f9cb..63cb1bc 100644
--- a/Gemfile
+++ b/Gemfile
@@ -4,7 +4,7 @@ gemspec name: 'jekyll'
 gem 'rake', '~> 10.1'
 group :development do
   gem 'rdoc', '~> 4.1'
-  gem 'toml', '=> 0.1.0'
+  gem 'toml', '~> 0.1.0'
   gem 'pry'
 end
 
