From: Youhei SASAKI <uwabami@gfd-dennou.org>
Date: Tue, 10 Jun 2014 16:32:44 +0900
Subject: Remove newline for newer Maruku

Signed-off-by: Youhei SASAKI <uwabami@gfd-dennou.org>
---
 test/test_tags.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/test/test_tags.rb
+++ b/test/test_tags.rb
@@ -525,7 +525,7 @@ CONTENT
 
       # todo: if #112 is merged into maruku, update to remove the newlines inside code block
       should "render fenced code blocks" do
-        assert_match %r{<pre class=\"ruby\"><code class=\"ruby\">\nputs &quot;Hello world&quot;\n</code></pre>}, @result.strip
+        assert_match %r{<pre class=\"ruby\"><code class=\"ruby\">puts &quot;Hello world&quot;</code></pre>}, @result.strip
       end
     end
 
