From: Youhei SASAKI <uwabami@gfd-dennou.org>
Date: Tue, 10 Jun 2014 17:31:51 +0900
Subject: Fix relative LOAD_PATH in test_redcloth

Signed-off-by: Youhei SASAKI <uwabami@gfd-dennou.org>
---
 test/test_redcloth.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/test/test_redcloth.rb
+++ b/test/test_redcloth.rb
@@ -1,4 +1,4 @@
-require File.dirname(__FILE__) + '/helper'
+require 'helper'
 
 class TestRedCloth < Test::Unit::TestCase
 
