From: Youhei SASAKI <uwabami@gfd-dennou.org>
Date: Tue, 10 Jun 2014 17:58:52 +0900
Subject: Add newline in test_coffee_script

Signed-off-by: Youhei SASAKI <uwabami@gfd-dennou.org>
---
 test/test_coffeescript.rb | 1 +
 1 file changed, 1 insertion(+)

--- a/test/test_coffeescript.rb
+++ b/test/test_coffeescript.rb
@@ -11,6 +11,7 @@ class TestCoffeeScript < Test::Unit::TestCase
       @test_coffeescript_file = dest_dir("js/coffeescript.js")
       @js_output = <<-JS
 (function() {
+
   $(function() {
     var cube, cubes, list, num, square;
     list = [1, 2, 3, 4, 5];
