From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Tue, 4 Aug 2015 23:53:53 +0200
Subject: Add Debian specific SONAME

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f34b0da..e787e20 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,6 +25,7 @@ include_directories(${CURL_INCLUDE_DIRS})
 
 add_library(${PROJECT_NAME} src/retriever.cpp)
 target_link_libraries(${PROJECT_NAME} ${CURL_LIBRARIES} ${catkin_LIBRARIES})
+set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${resource_retriever_VERSION} SOVERSION "0d")
 
 if(CATKIN_ENABLE_TESTING)
   add_subdirectory(test EXCLUDE_FROM_ALL)
