From b446c8ec531b4fb720afbe9ad501dce8413125f5 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 23 Jan 2020 14:27:02 -0600
Subject: [PATCH 2/2] drop these extra

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index f78f3e0..13778c3 100644
--- a/Makefile
+++ b/Makefile
@@ -27,6 +27,8 @@ endif
 install: ./bone101
 	install -m 0775 -d $(DESTDIR)$(PREFIX)/share/bone101
 	cp -dr --preserve=mode,timestamp ./bone101/* $(DESTDIR)$(PREFIX)/share/bone101/
+	rm -rf $(DESTDIR)$(PREFIX)/share/bone101/debian/ || true
+	rm -rf $(DESTDIR)$(PREFIX)/share/bone101/tmp/ || true
 	install -m 0775 -d $(DESTDIR)$(PREFIX)/share/applications
 	cp --preserve=mode,timestamp bone101.desktop $(DESTDIR)$(PREFIX)/share/applications/
 
-- 
2.24.1

