From c49241a0b55830330d8b268d08e45e1931215f5a Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 23 Jan 2020 14:21:54 -0600
Subject: [PATCH 2/2] drop these extra

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index f78f3e0..0698252 100644
--- a/Makefile
+++ b/Makefile
@@ -26,6 +26,8 @@ endif
 
 install: ./bone101
 	install -m 0775 -d $(DESTDIR)$(PREFIX)/share/bone101
+	rm -rf ./debian/ || true
+	rm -rf ./tmp/ || true
 	cp -dr --preserve=mode,timestamp ./bone101/* $(DESTDIR)$(PREFIX)/share/bone101/
 	install -m 0775 -d $(DESTDIR)$(PREFIX)/share/applications
 	cp --preserve=mode,timestamp bone101.desktop $(DESTDIR)$(PREFIX)/share/applications/
-- 
2.24.1

