diff --git a/pcmanfm/settings.cpp b/pcmanfm/settings.cpp
index dce0d3f..83d61f2 100644
--- a/pcmanfm/settings.cpp
+++ b/pcmanfm/settings.cpp
@@ -151,7 +151,7 @@ bool Settings::loadFile(QString filePath) {
     fallbackIconThemeName_ = "oxygen"; // fallback icon theme name
   }
   suCommand_ = settings.value("SuCommand", "gksu %s").toString();
-  setTerminal(settings.value("Terminal", "xterm").toString());
+  setTerminal(settings.value("Terminal", "qterminal").toString());
   setArchiver(settings.value("Archiver", "file-roller").toString());
   setSiUnit(settings.value("SIUnit", false).toBool());
 
