Index: llvm-toolchain-snapshot_3.5~svn199197/tools/llvm-shlib/Makefile
===================================================================
--- llvm-toolchain-3.4_3.4~svn209031.orig/tools/llvm-shlib/Makefile
+++ llvm-toolchain-3.4_3.4~svn209031/tools/llvm-shlib/Makefile
@@ -78,7 +78,7 @@ endif
 
 ifeq ($(HOST_OS), $(filter $(HOST_OS), DragonFly Linux FreeBSD GNU/kFreeBSD GNU))
     # Add soname to the library.
-    LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)$(SHLIBEXT) -Wl,-Bsymbolic-functions
+    LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)$(SHLIBEXT).1 -Wl,-Bsymbolic-functions
 endif
 
 ifeq ($(HOST_OS), $(filter $(HOST_OS), Linux GNU GNU/kFreeBSD))
