Description: Make sure PowerPC doesn't default to altivec on
Author: Adam Conrad <adconrad@ubuntu.com>
Author: Peter Michael Green <plugwash@debian.org>
Forwarded: no
Reviewed-By: Colin Watson <cjwatson@ubuntu.com>

Last-Update: 2013-04-01

Index: llvm-toolchain-snapshot_3.4~svn188990/clang/lib/Lex/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.4~svn188990.orig/clang/lib/Lex/Makefile	2013-08-22 09:11:26.616602695 +0200
+++ llvm-toolchain-snapshot_3.4~svn188990/clang/lib/Lex/Makefile	2013-08-22 09:11:26.616602695 +0200
@@ -16,11 +16,5 @@
 
 LIBRARYNAME := clangLex
 
-ifeq ($(ARCH),PowerPC)
-ifneq ($(shell dpkg-architecture -qDEB_HOST_ARCH),powerpcspe)
-CXX.Flags += -maltivec
-endif
-endif
-
 include $(CLANG_LEVEL)/Makefile
 
