Index: llvm-toolchain-3.5-3.5~+rc4/lldb/lib/Makefile
===================================================================
--- llvm-toolchain-3.5-3.5~+rc4.orig/lldb/lib/Makefile
+++ llvm-toolchain-3.5-3.5~+rc4/lldb/lib/Makefile
@@ -175,7 +175,7 @@ ifeq ($(HOST_OS), $(filter $(HOST_OS), L
     LLVMLibsOptions += -Wl,--no-undefined
     # Link in python
     LLVMLibsOptions += $(PYTHON_BUILD_FLAGS) -lrt -ledit -lncurses -lpanel
-    LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)$(SHLIBEXT).$(SONAME_EXTENSION)
+	LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)-$(LLVM_VERSION_MAJOR).$(LLVM_VERSION_MINOR)$(SHLIBEXT)
     ifneq (,$(filter $(shell dpkg-architecture -qDEB_HOST_ARCH), mips mipsel powerpc powerpcspe))
      # Link GCC atomic helper library
      LLVMLibsOptions += -latomic
