Index: llvm-toolchain-snapshot_3.5~svn211089/lldb/lib/Makefile
===================================================================
--- llvm-toolchain-snapshot_3.5~svn211089.orig/lldb/lib/Makefile
+++ llvm-toolchain-snapshot_3.5~svn211089/lldb/lib/Makefile
@@ -16,6 +16,7 @@ LIBRARYNAME = lldb
 NO_BUILD_ARCHIVE = 1
 LINK_LIBS_IN_SHARED = 1
 SHARED_LIBRARY = 1
+SONAME_EXTENSION = 1
 
 ifeq (,$(findstring -DLLDB_DISABLE_PYTHON,$(CXXFLAGS)))
 PYTHON_BUILD_FLAGS = $(shell python-config --ldflags)
@@ -169,7 +170,7 @@ ifeq ($(HOST_OS), $(filter $(HOST_OS), L
     LLVMLibsOptions += -Wl,--no-undefined
     # Link in python
     LLVMLibsOptions += $(PYTHON_BUILD_FLAGS) -lrt -ledit -lncurses -lpanel
-    LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)$(SHLIBEXT)
+    LLVMLibsOptions += -Wl,--soname,lib$(LIBRARYNAME)$(SHLIBEXT).$(SONAME_EXTENSION)
     ifneq (,$(filter $(shell dpkg-architecture -qDEB_HOST_ARCH), mips mipsel powerpc powerpcspe))
      # Link GCC atomic helper library
      LLVMLibsOptions += -latomic
