Index: llvm-toolchain-snapshot_3.5~svn199197/clang/bindings/python/clang/cindex.py
===================================================================
--- llvm-toolchain-snapshot_3.5~svn199197.orig/clang/bindings/python/clang/cindex.py	2014-01-14 11:31:18.444868130 +0100
+++ llvm-toolchain-snapshot_3.5~svn199197/clang/bindings/python/clang/cindex.py	2014-01-14 11:31:18.440868130 +0100
@@ -3371,7 +3371,7 @@
         elif name == 'Windows':
             file = 'libclang.dll'
         else:
-            file = 'libclang.so'
+            file = 'libclang-3.6.so'
 
         if Config.library_path:
             file = Config.library_path + '/' + file
