From 323567eacd52dc50997c063c8941bb57bbc107ff Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 27 Apr 2018 12:37:11 -0500
Subject: [PATCH] drop toolchain

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 Tools/ardupilotwaf/boards.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/Tools/ardupilotwaf/boards.py b/Tools/ardupilotwaf/boards.py
index 51c28d0..fcd9592 100644
--- a/Tools/ardupilotwaf/boards.py
+++ b/Tools/ardupilotwaf/boards.py
@@ -512,8 +512,6 @@ class ocpoc_zynq(linux):
         )
 
 class bbbmini(linux):
-    toolchain = 'arm-linux-gnueabihf'
-
     def configure_env(self, cfg, env):
         super(bbbmini, self).configure_env(cfg, env)
 
@@ -522,8 +520,6 @@ class bbbmini(linux):
         )
 
 class blue(linux):
-    toolchain = 'arm-linux-gnueabihf'
-
     def configure_env(self, cfg, env):
         super(blue, self).configure_env(cfg, env)
 
@@ -532,7 +528,6 @@ class blue(linux):
         )
 
 class pocket(linux):
-    toolchain = 'arm-linux-gnueabihf'
 
     def configure_env(self, cfg, env):
         super(pocket, self).configure_env(cfg, env)
-- 
2.17.0

