From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Tue, 9 Dec 2014 23:34:55 +0100
Subject: Add CMakeLists.txt

---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..e8e747a
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,5 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(ros-comm-msgs)
+
+add_subdirectory(rosgraph_msgs)
+add_subdirectory(std_srvs)
