Description: Fix typo in src/main_common.cpp
Author: Andrea Colangelo <warp10@debian.org>
Last-Update: 2014-02-01

--- xinput-calibrator-0.7.5+git20130410.orig/src/main_common.cpp
+++ xinput-calibrator-0.7.5+git20130410/src/main_common.cpp
@@ -197,7 +197,7 @@ static void usage(char* cmd, unsigned th
     fprintf(stderr, "\t--precalib: manually provide the current calibration setting (eg. the values in xorg.conf)\n");
     fprintf(stderr, "\t--misclick: set the misclick threshold (0=off, default: %i pixels)\n",
         thr_misclick);
-    fprintf(stderr, "\t--output-type <auto|xorg.conf.d|hal|xinput>: type of config to ouput (auto=automatically detect, default: auto)\n");
+    fprintf(stderr, "\t--output-type <auto|xorg.conf.d|hal|xinput>: type of config to output (auto=automatically detect, default: auto)\n");
     fprintf(stderr, "\t--fake: emulate a fake device (for testing purposes)\n");
     fprintf(stderr, "\t--geometry: manually provide the geometry (width and height) for the calibration window\n");
     fprintf(stderr, "\t--no-timeout: turns off the timeout\n");
