From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Wed, 5 Aug 2015 08:34:19 +0200
Subject: Add Debian specific SOVERSION

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a704301..45a7799 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,6 +17,7 @@ add_library(${PROJECT_NAME}
   src/random_numbers.cpp)
 
 target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES})
+set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${random_numbers_VERSION} SOVERSION "0d")
 
 install(TARGETS ${PROJECT_NAME}
         ARCHIVE DESTINATION lib
