From: Youhei SASAKI <uwabami@gfd-dennou.org>
Date: Wed, 11 Nov 2015 00:00:58 +0900
Subject: Replace gemspec git execution

Signed-off-by: Youhei SASAKI <uwabami@gfd-dennou.org>
---
 jekyll-coffeescript.gemspec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/jekyll-coffeescript.gemspec b/jekyll-coffeescript.gemspec
index ee5348b..871678e 100644
--- a/jekyll-coffeescript.gemspec
+++ b/jekyll-coffeescript.gemspec
@@ -12,7 +12,7 @@ Gem::Specification.new do |spec|
   spec.homepage      = "https://github.com/jekyll/jekyll-coffeescript"
   spec.license       = "MIT"
 
-  spec.files         = `git ls-files`.split($/).grep(%r{(lib/)})
+  spec.files         = [".gitignore",".rspec",".travis.yml","Gemfile","History.markdown","LICENSE.txt","README.md","Rakefile","jekyll-coffeescript.gemspec","lib/jekyll-coffeescript.rb","lib/jekyll-coffeescript/version.rb","lib/jekyll/converters/coffeescript.rb","script/bootstrap","script/cibuild","script/rebund","script/release","spec/coffeescript_spec.rb","spec/spec_helper.rb"]
   spec.require_paths = ["lib"]
 
   spec.add_runtime_dependency "coffee-script", "~> 2.2"
