Description: <short summary of the patch>
Author: Alf Gaida <agaida@siduction.org>

--- cmst-2016.01.28.orig/apps/cmstapp/code/control_box/controlbox.cpp
+++ cmst-2016.01.28/apps/cmstapp/code/control_box/controlbox.cpp
@@ -429,14 +429,11 @@ void ControlBox::aboutCMST()
                   "<center>sqozz (German)"
                   "<center>Ilya Shestopalov (Russian)"
                   "<br><center><b>Build Information:</b>"
-                  "<center>Compiled using QT version %5"
-                  "<center>Built %6 - %7")	.arg(TranslateStrings::cmtr("cmst"))
+                  "<center>Compiled using QT version %5")	.arg(TranslateStrings::cmtr("cmst"))
 																						.arg(VERSION)
 																						.arg(RELEASE_DATE)
 																						.arg(COPYRIGHT_DATE)
-																						.arg(QT_VERSION_STR)
-																						.arg(__DATE__)
-																						.arg(__TIME__) );
+																						.arg(QT_VERSION_STR));
 }

 //
