description: enable build hardening flags
author: Michael Gilbert <mgilbert@debian.org>

--- a/configure.py
+++ b/configure.py
@@ -85,6 +85,13 @@ llvm_cxxflags = llvm_config(['--cxxflags
                 '-DHAVE_LLVM=0x{:0=4}'.format(llvm_int_version)
 llvm_libdir = llvm_config(['--libdir'])
 
+# add in debian buildflags
+proc_ldflags = Popen(("dpkg-buildflags","--get","LDFLAGS"), stdout=PIPE)
+proc_cxxflags = Popen(("dpkg-buildflags","--get","CXXFLAGS"), stdout=PIPE)
+llvm_core_libs += " " + proc_ldflags.communicate()[0].strip("\n")
+llvm_cxxflags = llvm_cxxflags.replace(" -fPIC","")
+llvm_cxxflags += " " + proc_cxxflags.communicate()[0].strip("\n")
+
 llvm_clang = os.path.join(llvm_bindir, 'clang')
 llvm_as = os.path.join(llvm_bindir, 'llvm-as')
 llvm_link = os.path.join(llvm_bindir, 'llvm-link')
@@ -211,6 +218,13 @@ for target in targets:
                      "-emit-llvm" % (target, clang_cl_includes)
     if device['gpu'] != '':
       clang_bc_flags += ' -mcpu=' + device['gpu']
+
+    # add in debian build flags
+    proc_cflags = Popen(("dpkg-buildflags","--get","CFLAGS"), stdout=PIPE)
+    clang_bc_flags += " " + proc_cflags.communicate()[0].strip("\n")
+    # -fstack-protector-strong breaks some OpenCL apps, see Debian bug #904718
+    clang_bc_flags = clang_bc_flags.replace(" -fstack-protector-strong","")
+
     clang_bc_rule = "CLANG_CL_BC_" + target + "_" + device['gpu']
     c_compiler_rule(b, clang_bc_rule, "LLVM-CC", llvm_clang, clang_bc_flags)
     as_bc_rule = "LLVM_AS_BC_" + target + "_" + device['gpu']
--- a/build/metabuild.py
+++ b/build/metabuild.py
@@ -48,7 +48,7 @@ endif
 
     r = self.rules[rule]
     command = subst(r['command'])
-    if r['description']:
+    if False:
       desc = subst(r['description'])
       self.output.write('\t@echo %s\n\t$(Verb) %s\n' % (desc, command))
     else:
