From: Leopold Palomo-Avellaneda <leopold.palomo@upc.edu>
Date: Mon, 27 Apr 2015 15:38:55 +0200
Subject: Add-Debian-specific-SONAME

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fbb9920..98e939e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,6 +20,7 @@ catkin_package(
 )
 
 add_library(actionlib src/connection_monitor.cpp src/goal_id_generator.cpp)
+set_target_properties(actionlib PROPERTIES VERSION ${actionlib_VERSION} SOVERSION "0d")
 target_link_libraries(actionlib ${catkin_LIBRARIES} ${Boost_LIBRARIES})
 add_dependencies(actionlib actionlib_gencpp)
 
