--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -78,7 +78,7 @@ lightdm_CFLAGS = \
 	$(LIGHTDM_CFLAGS) \
 	-I"$(top_srcdir)/common" \
 	-DSBIN_DIR=\"$(sbindir)\" \
-	-DUSERS_DIR=\"$(localstatedir)/lib/lightdm-data\" \
+	-DUSERS_DIR=\"$(localstatedir)/lib/lightdm/data\" \
 	-DLOG_DIR=\"$(localstatedir)/log/lightdm\" \
 	-DRUN_DIR=\"$(localstatedir)/run/lightdm\" \
 	-DCACHE_DIR=\"$(localstatedir)/cache/lightdm\" \
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -476,7 +476,7 @@ lightdm_CFLAGS = \
 	$(LIGHTDM_CFLAGS) \
 	-I"$(top_srcdir)/common" \
 	-DSBIN_DIR=\"$(sbindir)\" \
-	-DUSERS_DIR=\"$(localstatedir)/lib/lightdm-data\" \
+	-DUSERS_DIR=\"$(localstatedir)/lib/lightdm/data\" \
 	-DLOG_DIR=\"$(localstatedir)/log/lightdm\" \
 	-DRUN_DIR=\"$(localstatedir)/run/lightdm\" \
 	-DCACHE_DIR=\"$(localstatedir)/cache/lightdm\" \
--- a/src/shared-data-manager.c
+++ b/src/shared-data-manager.c
@@ -99,7 +99,7 @@ shared_data_manager_ensure_user_dir (Sha
 
     g_debug ("Creating shared data directory %s", path);
 
-    result = g_file_make_directory (file, NULL, &error);
+    result = g_file_make_directory_with_parents (file, NULL, &error);
     if (error)
     {
         if (g_error_matches (error, G_IO_ERROR, G_IO_ERROR_EXISTS))
