diff -urN node-v0.11.13-orig/common.gypi node-v0.11.13/common.gypi
--- node-v0.11.13-orig/common.gypi	2014-05-01 20:49:51.000000000 -0400
+++ node-v0.11.13/common.gypi	2014-06-12 00:58:50.839136495 -0400
@@ -11,6 +11,7 @@
     'gcc_version%': 'unknown',
     'clang%': 0,
     'python%': 'python',
+    'arm_version%': '1',
 
     # Enable V8's post-mortem debugging only on unix flavors.
     'conditions': [
