diff -urN nodejs-0.8.0.orig/tools/node-waf nodejs-0.8.0/tools/node-waf
--- nodejs-0.8.0.orig/tools/node-waf	2012-06-25 14:37:20.000000000 +0000
+++ nodejs-0.8.0/tools/node-waf	2012-06-27 20:32:48.000000000 +0000
@@ -4,8 +4,8 @@
 
 join = os.path.join
 bindir = os.path.dirname(os.path.realpath(__file__))
-prefix = join(bindir, "..")
-wafdir = join(prefix, "lib", "node")
+prefix = join('/usr')
+wafdir = join(prefix, "share", "nodejs")
 
 w = join(wafdir, 'wafadmin')
 t = join(w, 'Tools')
