diff -urpN quilt-0.63.orig/quilt/header.in quilt-0.63/quilt/header.in
--- quilt-0.63.orig/quilt/header.in	2016-03-07 08:54:59.000000000 -0700
+++ quilt-0.63/quilt/header.in	2016-03-07 09:50:05.864357562 -0700
@@ -29,7 +29,7 @@ usage()
 Print or change the header of the topmost or specified patch.
 
 -a, -r, -e
-	Append to (-a) or replace (-r) the exiting patch header, or
+	Append to (-a) or replace (-r) the existing patch header, or
 	edit (-e) the header in \$EDITOR (%s). If none of these options is
 	given, print the patch header.
 
