--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -105,7 +105,7 @@
     add_subdirectory(plugin-colorpicker)
 endif()
 
-setByDefault(CPULOAD_PLUGIN Yes)
+setByDefault(CPULOAD_PLUGIN Yes)
 if(CPULOAD_PLUGIN)
     find_library(STATGRAB_LIB statgrab)
 
@@ -195,7 +195,7 @@
     add_subdirectory(plugin-showdesktop)
 endif()
 
-setByDefault(NETWORKMONITOR_PLUGIN Yes)
+setByDefault(NETWORKMONITOR_PLUGIN Yes)
 if(NETWORKMONITOR_PLUGIN)
     find_library(STATGRAB_LIB statgrab)
 
@@ -209,7 +209,7 @@
     endif()
 endif()
 
-setByDefault(SYSSTAT_PLUGIN Yes)
+setByDefault(SYSSTAT_PLUGIN Yes)
 if(SYSSTAT_PLUGIN)
     set(ENABLED_PLUGINS ${ENABLED_PLUGINS} "System Stats")
     add_subdirectory(plugin-sysstat)
