From 3be7a92c2269eb4e0c0706289e761904ee3d773a Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 8 Aug 2018 10:29:19 -0500
Subject: [PATCH] aet: use cgt-c6x include path

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 aet/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aet/Makefile b/aet/Makefile
index 0d3ad95..345f52c 100644
--- a/aet/Makefile
+++ b/aet/Makefile
@@ -11,7 +11,7 @@ endif
 
 
 CFLAGS=-o3 $(TARGET_CFLAGS) --abi=eabi --mem_model:data=far -eoo \
-        --visibility=protected -Iinclude
+        --visibility=protected -Iinclude -I=/usr/share/ti/cgt-c6x/include/
 
 VPATH = src
 
-- 
2.18.0

