Index: qtcreator/src/plugins/coreplugin/icore.cpp
===================================================================
--- qtcreator.orig/src/plugins/coreplugin/icore.cpp	2014-12-08 06:58:49.000000000 -0600
+++ qtcreator/src/plugins/coreplugin/icore.cpp	2015-01-01 02:56:43.042709789 -0600
@@ -422,7 +422,7 @@
 QString ICore::documentationPath()
 {
     const QString docPath = QLatin1String(Utils::HostOsInfo::isMacHost()
-                                            ? "/../Resources/doc" : "/../share/doc/qtcreator");
+                                            ? "/../Resources/doc" : "/../share/qtcreator/doc");
     return QDir::cleanPath(QCoreApplication::applicationDirPath() + docPath);
 }
 
