Index: qtcreator/src/libs/ssh/sshcryptofacility.cpp
===================================================================
--- qtcreator.orig/src/libs/ssh/sshcryptofacility.cpp	2014-12-25 11:28:21.388740421 -0600
+++ qtcreator/src/libs/ssh/sshcryptofacility.cpp	2014-12-25 22:10:30.002238360 -0600
@@ -39,6 +39,7 @@
 
 #include <botan/botan.h>
 #include <botan/cbc.h>
+#include <botan/ctr.h>
 #include <botan/dsa.h>
 #include <botan/pkcs8.h>
 #include <botan/pubkey.h>
