From f345d9e48c9e1169edf047de742da142cc5687bc Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Mon, 23 Sep 2013 11:05:12 -0500
Subject: [PATCH 2/2] dtc: v takes no argument drop extra :

$ git show dd6a0533
... snip ...
@@ -49,7 +50,7 @@ static void fill_fullpaths(struct node *tree, const char *prefix)

 /* Usage related data. */
 static const char usage_synopsis[] = "dtc [options] <input file>";
-static const char usage_short_opts[] = "qI:O:o:V:d:R:S:p:fb:i:H:sW:E:hv";
+static const char usage_short_opts[] = "qI:O:o:V:d:R:S:p:fb:i:H:sW:E:hv:@";
 static struct option const usage_long_opts[] = {
        {"quiet",            no_argument, NULL, 'q'},
        {"in-format",         a_argument, NULL, 'I'},

  that patch is wrong, there should be no ":" after "v"

Reported-by: Robert P. J. Day <rpjday@crashcourse.ca>
Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 dtc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dtc.c b/dtc.c
index d2f9647..b12ec2d 100644
--- a/dtc.c
+++ b/dtc.c
@@ -50,7 +50,7 @@ static void fill_fullpaths(struct node *tree, const char *prefix)
 
 /* Usage related data. */
 static const char usage_synopsis[] = "dtc [options] <input file>";
-static const char usage_short_opts[] = "qI:O:o:V:d:R:S:p:fb:i:H:sW:E:hv:@";
+static const char usage_short_opts[] = "qI:O:o:V:d:R:S:p:fb:i:H:sW:E:hv@";
 static struct option const usage_long_opts[] = {
 	{"quiet",            no_argument, NULL, 'q'},
 	{"in-format",         a_argument, NULL, 'I'},
-- 
1.7.10.4

