From: Alexander Sack <asac@debian.org>
Subject: add missing --nobacktrace option to manpage
Bug-Debian: http://bugs.debian.org/822610
Forwarded: https://lists.01.org/pipermail/connman/2016-September/020878.html
Last-Updated: 2016-09-06
---
diff --git a/doc/connman.8.in b/doc/connman.8.in
index 9b71779..07e6ef5 100644
--- a/doc/connman.8.in
+++ b/doc/connman.8.in
@@ -80,6 +80,9 @@
 Wifi driver that WiFi/Supplicant should use. If this flag is omitted,
 then the value "nl80211,wext" is used by default.
 .TP
+.BR \-n ", " \-\-nobacktrace
+Don't print out backtrace information.
+.TP
 .BR \-n ", " \-\-nodaemon
 Do not daemonize. This is useful for debugging, and directs log output to
 the controlling terminal in addition to syslog.

