From 9221feeaf97d1a91f317acd428ad2242a23761a3 Mon Sep 17 00:00:00 2001
From: Jason Kridner <jdk@ti.com>
Date: Thu, 1 Mar 2018 11:41:08 -0500
Subject: [PATCH] Disable roboticscape for now

---
 src/index.js | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/index.js b/src/index.js
index 048c04eb..5689264f 100644
--- a/src/index.js
+++ b/src/index.js
@@ -17,7 +17,7 @@ var epoll = my.require('epoll');
 var autorun = require('./autorun');
 var server = require('./server');
 var socketHandlers = require('./socket_handlers');
-var rc = require('./rc');
+//var rc = require('./rc');
 
 var debug = process.env.DEBUG ? true : false;
 
@@ -585,9 +585,9 @@ for(var x in server) {
 for(var x in socketHandlers) {
     exports[x] = socketHandlers[x];
 }
-for(var x in rc) {
+/*for(var x in rc) {
     exports[x] = rc[x];
-}
+}*/
 
 var alreadyRan = false;
 function run() {
