diff --git a/dart/runtime/BUILD.gn b/dart/runtime/BUILD.gn
index f2d15cb..8e3b2b0 100644
--- a/dart/runtime/BUILD.gn
+++ b/dart/runtime/BUILD.gn
@@ -61,6 +61,7 @@ config("dart_config") {
     "-Wno-unused-parameter",
     "-Wno-unused-variable",
     "-Wno-strict-overflow",
+    "-Wno-shift-negative-value",
     "-Wnon-virtual-dtor",
     "-Wvla",
     "-Wno-conversion-null",
diff --git a/dart/tools/gyp/configurations.gypi b/dart/tools/gyp/configurations.gypi
index 40a3c9e..b7da16b 100644
--- a/dart/tools/gyp/configurations.gypi
+++ b/dart/tools/gyp/configurations.gypi
@@ -10,6 +10,7 @@
       '-Wno-unused-parameter',
       '-Wno-unused-variable',
       '-Wno-strict-overflow',
+      '-Wno-shift-negative-value',
     ],
 
     # Default value.  This may be overridden in a containing project gyp.
