--- a/scripts/connman.in	2012-05-10 14:29:21.020396338 +0300
+++ b/scripts/connman.in	2012-05-10 14:29:49.199127827 +0300
@@ -1,4 +1,13 @@
 #!/bin/sh
+### BEGIN INIT INFO
+# Provides:       connman
+# Required-Start: $local_fs $syslog
+# Required-Stop:  $local_fs $syslog
+# Default-Start:  2 3 4 5
+# Default-Stop:   0 1 6
+# Short-Description: Connection Manager
+### END INIT INFO
+
 
 DAEMON=@prefix@/sbin/connmand
 DESC="Connection Manager"
