From ba41ded30e004f466412973bea84fdc60f61869f Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 6 Oct 2015 16:04:48 -0500
Subject: [PATCH] scripts/functions: add PARTUUID to resolve_device function

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 init | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/init b/init
index cb832ff..2837f68 100755
--- a/init
+++ b/init
@@ -97,6 +97,9 @@ for x in $(cat /proc/cmdline); do
 			esac
 			ROOT="/dev/disk/by-label/${ROOT}"
 			;;
+		PARTUUID=*)
+			ROOT="/dev/disk/by-partuuid/${ROOT#PARTUUID=}"
+			;;
 		UUID=*)
 			ROOT="/dev/disk/by-uuid/${ROOT#UUID=}"
 			;;
-- 
2.5.3

