--- openjdk-boot/jdk/src/share/classes/java/lang/Float.java~	2010-03-08 14:53:10.224835876 +0100
+++ openjdk-boot/jdk/src/share/classes/java/lang/Float.java	2010-03-08 15:05:28.803585964 +0100
@@ -86,7 +86,7 @@
      *
      * @since 1.6
      */
-    public static final float MIN_NORMAL = 1.17549435E-38f;
+    public static final float MIN_NORMAL = Float.intBitsToFloat(0x00800000);
 
     /**
      * A constant holding the smallest positive nonzero value of type
