--- openjdk/jdk/src/solaris/native/java/net/NetworkInterface.c~	2012-02-15 09:26:50.000000000 +0100
+++ openjdk/jdk/src/solaris/native/java/net/NetworkInterface.c	2012-02-26 14:13:22.602107361 +0100
@@ -1105,7 +1105,7 @@
     uint8_t ipv6addr[16];
 
     if ((f = fopen(_PATH_PROCNET_IFINET6, "r")) != NULL) {
-        while (fscanf(f, "%4s%4s%4s%4s%4s%4s%4s%4s %02x %02x %02x %02x %20s\n",
+        while (fscanf(f, "%4s%4s%4s%4s%4s%4s%4s%4s %x %x %x %x %20s\n",
                          addr6p[0], addr6p[1], addr6p[2], addr6p[3], addr6p[4], addr6p[5], addr6p[6], addr6p[7],
                          &if_idx, &plen, &scope, &dad_status, devname) != EOF) {
 
