# DP: Turn off -Werror for hotspot, overwrites -fpermissive.

--- src/hotspot/make/linux/makefiles/gcc.make~	2014-07-02 15:25:35.805646895 +0200
+++ src/hotspot/make/linux/makefiles/gcc.make	2014-07-02 15:37:59.346690749 +0200
@@ -205,7 +205,7 @@
 endif
 
 # Compiler warnings are treated as errors
-WARNINGS_ARE_ERRORS = -Werror -Wno-error=format -Wno-error=deprecated
+WARNINGS_ARE_ERRORS = 
 
 ifeq ($(USE_CLANG), true)
   # However we need to clean the code up before we can unrestrictedly enable this option with Clang
