--- openjdk/hotspot/src/share/vm/oops/arrayKlassKlass.cpp~	2013-03-04 22:51:00.000000000 +0100
+++ openjdk/hotspot/src/share/vm/oops/arrayKlassKlass.cpp	2013-04-15 13:57:24.992929086 +0200
@@ -30,6 +30,7 @@
 #include "runtime/handles.inline.hpp"
 #ifndef SERIALGC
 #include "gc_implementation/parNew/parOopClosures.inline.hpp"
+#include "gc_implementation/parallelScavenge/psOldGen.hpp"
 #include "gc_implementation/parallelScavenge/psPromotionManager.inline.hpp"
 #include "gc_implementation/parallelScavenge/psScavenge.inline.hpp"
 #include "memory/cardTableRS.hpp"
--- openjdk/hotspot/src/share/vm/oops/constantPoolKlass.cpp~	2013-03-04 22:51:00.000000000 +0100
+++ openjdk/hotspot/src/share/vm/oops/constantPoolKlass.cpp	2013-04-15 14:02:52.143087149 +0200
@@ -49,6 +49,7 @@
 #endif
 #ifndef SERIALGC
 #include "gc_implementation/parNew/parOopClosures.inline.hpp"
+#include "gc_implementation/parallelScavenge/psOldGen.hpp"
 #include "gc_implementation/parallelScavenge/psPromotionManager.inline.hpp"
 #include "gc_implementation/parallelScavenge/psScavenge.inline.hpp"
 #include "memory/cardTableRS.hpp"
--- openjdk/hotspot/src/share/vm/oops/cpCacheKlass.cpp~	2013-03-04 22:51:00.000000000 +0100
+++ openjdk/hotspot/src/share/vm/oops/cpCacheKlass.cpp	2013-04-15 14:04:45.219838633 +0200
@@ -35,6 +35,7 @@
 #include "runtime/handles.inline.hpp"
 #ifndef SERIALGC
 #include "gc_implementation/parNew/parOopClosures.inline.hpp"
+#include "gc_implementation/parallelScavenge/psOldGen.hpp"
 #include "gc_implementation/parallelScavenge/psPromotionManager.inline.hpp"
 #include "gc_implementation/parallelScavenge/psScavenge.inline.hpp"
 #include "memory/cardTableRS.hpp"
--- openjdk/hotspot/src/share/vm/oops/instanceKlass.cpp~	2013-03-04 22:51:00.000000000 +0100
+++ openjdk/hotspot/src/share/vm/oops/instanceKlass.cpp	2013-04-15 14:07:55.861109867 +0200
@@ -70,6 +70,7 @@
 #include "gc_implementation/g1/g1RemSet.inline.hpp"
 #include "gc_implementation/g1/heapRegionSeq.inline.hpp"
 #include "gc_implementation/parNew/parOopClosures.inline.hpp"
+#include "gc_implementation/parallelScavenge/psOldGen.hpp"
 #include "gc_implementation/parallelScavenge/psPromotionManager.inline.hpp"
 #include "gc_implementation/parallelScavenge/psScavenge.inline.hpp"
 #include "oops/oop.pcgc.inline.hpp"
--- openjdk/hotspot/src/share/vm/oops/instanceKlassKlass.cpp~	2013-03-04 22:51:00.000000000 +0100
+++ openjdk/hotspot/src/share/vm/oops/instanceKlassKlass.cpp	2013-04-15 14:08:53.541485528 +0200
@@ -44,6 +44,7 @@
 #include "runtime/fieldDescriptor.hpp"
 #ifndef SERIALGC
 #include "gc_implementation/parNew/parOopClosures.inline.hpp"
+#include "gc_implementation/parallelScavenge/psOldGen.hpp"
 #include "gc_implementation/parallelScavenge/psPromotionManager.inline.hpp"
 #include "gc_implementation/parallelScavenge/psScavenge.inline.hpp"
 #include "memory/cardTableRS.hpp"
--- openjdk/hotspot/src/share/vm/oops/instanceMirrorKlass.cpp~	2013-03-04 22:51:00.000000000 +0100
+++ openjdk/hotspot/src/share/vm/oops/instanceMirrorKlass.cpp	2013-04-15 14:10:11.162003798 +0200
@@ -42,6 +42,7 @@
 #include "gc_implementation/g1/g1RemSet.inline.hpp"
 #include "gc_implementation/g1/heapRegionSeq.inline.hpp"
 #include "gc_implementation/parNew/parOopClosures.inline.hpp"
+#include "gc_implementation/parallelScavenge/psOldGen.hpp"
 #include "gc_implementation/parallelScavenge/psPromotionManager.inline.hpp"
 #include "gc_implementation/parallelScavenge/psScavenge.inline.hpp"
 #include "oops/oop.pcgc.inline.hpp"
--- openjdk/hotspot/src/share/vm/oops/instanceRefKlass.cpp~	2013-03-04 22:51:00.000000000 +0100
+++ openjdk/hotspot/src/share/vm/oops/instanceRefKlass.cpp	2013-04-15 14:11:11.726403390 +0200
@@ -39,6 +39,7 @@
 #include "gc_implementation/g1/g1RemSet.inline.hpp"
 #include "gc_implementation/g1/heapRegionSeq.inline.hpp"
 #include "gc_implementation/parNew/parOopClosures.inline.hpp"
+#include "gc_implementation/parallelScavenge/psOldGen.hpp"
 #include "gc_implementation/parallelScavenge/psPromotionManager.inline.hpp"
 #include "gc_implementation/parallelScavenge/psScavenge.inline.hpp"
 #include "oops/oop.pcgc.inline.hpp"
--- openjdk/hotspot/src/share/vm/oops/klassKlass.cpp~	2013-03-04 22:51:00.000000000 +0100
+++ openjdk/hotspot/src/share/vm/oops/klassKlass.cpp	2013-04-15 14:13:01.999133769 +0200
@@ -42,6 +42,7 @@
 #include "runtime/handles.inline.hpp"
 #ifndef SERIALGC
 #include "gc_implementation/parNew/parOopClosures.inline.hpp"
+#include "gc_implementation/parallelScavenge/psOldGen.hpp"
 #include "gc_implementation/parallelScavenge/psPromotionManager.inline.hpp"
 #include "gc_implementation/parallelScavenge/psScavenge.inline.hpp"
 #include "memory/cardTableRS.hpp"
--- openjdk/hotspot/src/share/vm/oops/objArrayKlass.cpp~	2013-03-04 22:51:00.000000000 +0100
+++ openjdk/hotspot/src/share/vm/oops/objArrayKlass.cpp	2013-04-15 14:16:10.616389484 +0200
@@ -47,6 +47,7 @@
 #include "gc_implementation/g1/g1RemSet.inline.hpp"
 #include "gc_implementation/g1/heapRegionSeq.inline.hpp"
 #include "gc_implementation/parNew/parOopClosures.inline.hpp"
+#include "gc_implementation/parallelScavenge/psOldGen.hpp"
 #include "gc_implementation/parallelScavenge/psCompactionManager.hpp"
 #include "gc_implementation/parallelScavenge/psPromotionManager.inline.hpp"
 #include "gc_implementation/parallelScavenge/psScavenge.inline.hpp"
--- openjdk/hotspot/src/share/vm/oops/objArrayKlassKlass.cpp~	2013-03-04 22:51:00.000000000 +0100
+++ openjdk/hotspot/src/share/vm/oops/objArrayKlassKlass.cpp	2013-04-15 14:17:01.800729154 +0200
@@ -33,6 +33,7 @@
 #include "oops/oop.inline2.hpp"
 #ifndef SERIALGC
 #include "gc_implementation/parNew/parOopClosures.inline.hpp"
+#include "gc_implementation/parallelScavenge/psOldGen.hpp"
 #include "gc_implementation/parallelScavenge/psPromotionManager.inline.hpp"
 #include "gc_implementation/parallelScavenge/psScavenge.inline.hpp"
 #include "memory/cardTableRS.hpp"
--- openjdk/hotspot/src/share/vm/gc_implementation/parallelScavenge/psTasks.cpp~	2013-03-04 22:51:00.000000000 +0100
+++ openjdk/hotspot/src/share/vm/gc_implementation/parallelScavenge/psTasks.cpp	2013-04-15 14:20:24.782079605 +0200
@@ -28,6 +28,7 @@
 #include "gc_implementation/parallelScavenge/cardTableExtension.hpp"
 #include "gc_implementation/parallelScavenge/gcTaskManager.hpp"
 #include "gc_implementation/parallelScavenge/psMarkSweep.hpp"
+#include "gc_implementation/parallelScavenge/psOldGen.hpp"
 #include "gc_implementation/parallelScavenge/psPromotionManager.hpp"
 #include "gc_implementation/parallelScavenge/psPromotionManager.inline.hpp"
 #include "gc_implementation/parallelScavenge/psScavenge.inline.hpp"
