From: Leopold Palomo-Avellaneda <leopold.palomo@upc.edu>
Date: Wed, 29 Apr 2015 13:12:38 +0200
Subject: Added CMakeLists to build the package

---
 CMakeLists.txt | 12 ++++++++++++
 1 file changed, 12 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..a03ca7a
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(geometry-experimental)
+add_subdirectory( tf2_msgs )
+add_subdirectory( tf2 )
+add_subdirectory( tf2_py )
+add_subdirectory( tf2_ros)
+add_subdirectory( tf2_bullet )
+add_subdirectory( tf2_eigen )
+add_subdirectory( tf2_geometry_msgs )
+add_subdirectory( tf2_kdl )
+add_subdirectory( tf2_sensor_msgs )
+add_subdirectory( tf2_tools )
