Description: do not build cctest, build broken on debian
Last-Update: 2017-12-18
Author: Jérémy Lal <kapouer@melix.org>
Forwarded: not yet !
--- a/Makefile
+++ b/Makefile
@@ -61,6 +61,10 @@
 # or set the V environment variable to an empty string.
 V ?= 1
 
+build:
+	$(MAKE) $(PARALLEL_ARGS) -C out $(NODE_EXE) BUILDTYPE=$(BUILDTYPE) V=$(V)
+	ln -sf out/$(BUILDTYPE)/node .
+
 # BUILDTYPE=Debug builds both release and debug builds. If you want to compile
 # just the debug build, run `make -C out BUILDTYPE=Debug` instead.
 ifeq ($(BUILDTYPE),Release)
@@ -101,7 +105,7 @@
 config.gypi: configure
 	$(error Missing or stale $@, please run ./$<)
 
-install: all ## Installs node into $PREFIX (default=/usr/local).
+install:
 	$(PYTHON) tools/install.py $@ '$(DESTDIR)' '$(PREFIX)'
 
 uninstall: ## Uninstalls node from $PREFIX (default=/usr/local).
