diff --git a/monitor/Makefile.am57x b/monitor/Makefile.am57x
index 262ae94..5bae693 100644
--- a/monitor/Makefile.am57x
+++ b/monitor/Makefile.am57x
@@ -40,7 +40,7 @@ DEVICE    = DEVICE_AM572x
 SOC       = SOC_AM572x
 
 ## !Need -O* to inline CACHE_* functions!
-CC= $(TI_OCL_CGT_INSTALL)/bin/cl6x -o3 -mv6600 --abi=eabi --gcc --mem_model:const=data --mem_model:data=far
+CC= cl6x -o3 -mv6600 --abi=eabi --gcc --mem_model:const=data --mem_model:data=far
 
 INCLUDES  = -i. -i $(OPENCL_SRC_DIR)/src/core/dsp
 INCLUDES += -i $(AM57_PDK_DIR)/packages -i $(TI_OCL_CGT_INSTALL)/include \
