Description: lazy load auth modules - they can be optional
Forwarded: https://github.com/mikeal/request/pull/628
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2013-08-14
--- a/index.js
+++ b/index.js
@@ -22,10 +22,10 @@
   , querystring = require('querystring')
   , crypto = require('crypto')
 
-  , oauth = require('oauth-sign')
-  , hawk = require('hawk')
-  , aws = require('aws-sign')
-  , httpSignature = require('http-signature')
+  , oauth         // lazy load
+  , hawk          // lazy load
+  , aws           // lazy load
+  , httpSignature // lazy load
   , uuid = require('node-uuid')
   , mime = require('mime')
   , tunnel = require('tunnel-agent')
@@ -287,18 +287,22 @@
 
   // Auth must happen last in case signing is dependent on other headers
   if (options.oauth) {
+    if (!oauth) oauth = require('oauth-sign');
     self.oauth(options.oauth)
   }
 
   if (options.aws) {
+    if (!aws) aws = require('aws-sign');
     self.aws(options.aws)
   }
 
   if (options.hawk) {
+    if (!hawk) hawk = require('hawk');
     self.hawk(options.hawk)
   }
 
   if (options.httpSignature) {
+    if (!httpSignature) httpSignature = require('http-signature');
     self.httpSignature(options.httpSignature)
   }
 
