From: Jochen Sprickerhof <jochen@sprickerhof.de>
Date: Mon, 24 Jun 2013 18:00:12 +0200
Subject: Add CMakeLists.txt

---
 CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..3bb2dfe
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(roscpp_core)
+
+add_subdirectory(roscpp_traits)
+add_subdirectory(cpp_common)
+add_subdirectory(rostime)
+add_subdirectory(roscpp_core)
+add_subdirectory(roscpp_serialization)
