From 3e1d7630187dd96b137553aec2c6edb60c213682
From: Tim Philipp <tim@centricular.com>
Date: Sat, 15 Nov 2014 21:59:48 +0000
Subject: dvbbasebin: fix possible crash by passing 64 bits for 64-bit queue
 property

https://bugzilla.gnome.org/show_bug.cgi?id=740191

--- a/sys/dvb/dvbbasebin.c
+++ b/sys/dvb/dvbbasebin.c
@@ -417,7 +417,7 @@ dvb_base_bin_init (DvbBaseBin * dvbbaseb
   dvbbasebin->tsparse = gst_element_factory_make ("tsparse", NULL);
 
   g_object_set (dvbbasebin->buffer_queue, "max-size-buffers", 0,
-      "max-size-bytes", 0, "max-size-time", 0, NULL);
+      "max-size-bytes", 0, "max-size-time", (guint64) 0, NULL);
 
   gst_bin_add_many (GST_BIN (dvbbasebin), dvbbasebin->dvbsrc,
       dvbbasebin->buffer_queue, dvbbasebin->tsparse, NULL);
