From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Thu, 2 Apr 2015 11:13:21 +0200
Subject: Fix location of /etc for catkin_find

---
 python/catkin/find_in_workspaces.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/python/catkin/find_in_workspaces.py b/python/catkin/find_in_workspaces.py
index 0c74f0a..8ef3ad6 100644
--- a/python/catkin/find_in_workspaces.py
+++ b/python/catkin/find_in_workspaces.py
@@ -123,6 +123,8 @@ def find_in_workspaces(search_dirs=None, project=None, path=None, _workspaces=ge
             for sub in search_dirs:
                 # search in workspace
                 p = os.path.join(workspace, sub)
+                if p == '/usr/etc':
+                    p = '/etc'
                 if project:
                     p = os.path.join(p, project)
                 if path:
