Index: qupzilla-1.8.6~dfsg1/src/defines.pri
===================================================================
--- qupzilla-1.8.6~dfsg1.orig/src/defines.pri
+++ qupzilla-1.8.6~dfsg1/src/defines.pri
@@ -5,6 +5,13 @@ MOC_DIR = $$PWD/../build
 RCC_DIR = $$PWD/../build
 UI_DIR = $$PWD/../build
 
+############## hardening flags for Debian ###############
+QMAKE_CPPFLAGS *= $(shell dpkg-buildflags --get CPPFLAGS)
+QMAKE_CFLAGS   *= $(shell dpkg-buildflags --get CFLAGS)
+QMAKE_CXXFLAGS *= $(shell dpkg-buildflags --get CXXFLAGS)
+QMAKE_LFLAGS   *= $(shell dpkg-buildflags --get LDFLAGS)
+
+
 # workaround for #849: see https://bugreports.qt-project.org/browse/QTBUG-23196
 mocinclude.CONFIG *= fix_target
 
