From 9de120704642bd2071a1b809035d03e652463a20 Mon Sep 17 00:00:00 2001
From: Thierry Reding <thierry.reding@gmail.com>
Date: Tue, 25 Aug 2015 17:01:02 +0200
Subject: [PATCH] libdrm: omap: fix lockups on am57xx

Tested-by: Robert Nelson <robertcnelson@gmail.com>
---
 omap/omap_drm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/omap/omap_drm.c b/omap/omap_drm.c
index 4a0248d..08ba64e 100644
--- a/omap/omap_drm.c
+++ b/omap/omap_drm.c
@@ -186,6 +186,7 @@ static struct omap_bo * bo_from_handle(struct omap_device *dev,
 	}
 	bo->dev = omap_device_ref(dev);
 	bo->handle = handle;
+	bo->fd = -1;
 	atomic_set(&bo->refcnt, 1);
 	/* add ourselves to the handle table: */
 	drmHashInsert(dev->handle_table, handle, bo);
-- 
2.5.0

