Description: Fix OnlyShowIn
Author: Alf Gaida <agaida@siduction.org>
Last-Update: 2014-10-18

--- lxqt-globalkeys-0.8.0.orig/config/lxqt-config-globalkeyshortcuts.desktop.in
+++ lxqt-globalkeys-0.8.0/config/lxqt-config-globalkeyshortcuts.desktop.in
@@ -6,6 +6,6 @@ Comment=Configure shortcut keys in LXQt
 Exec=lxqt-config-globalkeyshortcuts
 Icon=preferences-desktop-keyboard
 Categories=Settings;DesktopSettings;Qt;LXQt;
-OnlyShowIn=LXQt;LXDE;
+OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
