diff --git a/configure-device.sh b/configure-device.sh
index 52a32dd..63b1a72 100755
--- a/configure-device.sh
+++ b/configure-device.sh
@@ -1,13 +1,21 @@
-#!/bin/sh
+#!/bin/bash
 # Script to run on the target to configure the wl18xx-conf.bin file to match the device capabilities
 
+if ! [ "$(id -u)" = 0 ] ; then
+	echo "$0 must be run as sudo user or root"
+	exit 1
+fi
+
 # version
 VERSION=1.2
 
 # defaults
 binary_name="/lib/firmware/ti-connectivity/wl18xx-conf.bin"
-wlconf_path="/usr/sbin/wlconf/"
-ini_path="/usr/sbin/wlconf/official_inis"
+ini_path="/usr/share/ti/wlconf/official_inis"
+
+if [ ! -f ${binary_name} ] ; then
+	cp /usr/share/ti/wlconf/wl18xx-conf-default.bin ${binary_name}
+fi
 
 # function for printing help
 print_help()
@@ -25,7 +33,7 @@ print_help()
 print_dump()
 {
 	echo "Saving wl18xx-conf.bin dump to wl18xx-conf-dump.txt"
-	./wlconf -i $binary_name --get > wl18xx-conf-dump.txt
+	wlconf -i $binary_name --get > wl18xx-conf-dump.txt
 }
 
 # function for printing summary
@@ -253,15 +261,14 @@ rmmod wlcore_sdio
 
 
 # create clean file
-cd $wlconf_path
-./wlconf -o $binary_name -I $ini_file_name
+wlconf -o $binary_name -I $ini_file_name
   
   
 # finally write values to binary
-./wlconf -i $binary_name -o $binary_name -s wl18xx.phy.number_of_assembled_ant2_4=$number_2_4G_antenna
-./wlconf -i $binary_name -o $binary_name -s wl18xx.phy.number_of_assembled_ant5=$number_5G_antenna
-./wlconf -i $binary_name -o $binary_name -s wl18xx.phy.high_band_component_type=$high_band_component_type
-./wlconf -i $binary_name -o $binary_name -s wl18xx.ht.mode=$ht_mode
+wlconf -i $binary_name -o $binary_name -s wl18xx.phy.number_of_assembled_ant2_4=$number_2_4G_antenna
+wlconf -i $binary_name -o $binary_name -s wl18xx.phy.number_of_assembled_ant5=$number_5G_antenna
+wlconf -i $binary_name -o $binary_name -s wl18xx.phy.high_band_component_type=$high_band_component_type
+wlconf -i $binary_name -o $binary_name -s wl18xx.ht.mode=$ht_mode
 
 print_summary
 
diff --git a/wlconf.h b/wlconf.h
index 9999fd9..74ab9f8 100644
--- a/wlconf.h
+++ b/wlconf.h
@@ -68,7 +68,7 @@ struct type types[] = {
  * See in Android.mk
  */
 #ifndef WLCONF_DIR
-#define WLCONF_DIR ""
+#define WLCONF_DIR "/usr/share/ti/wlconf/"
 #endif
 
 #define DEFAULT_INPUT_FILENAME	WLCONF_DIR "wl18xx-conf-default.bin"
