Index: pastebinit/pastebinit
===================================================================
--- pastebinit.orig/pastebinit	2014-01-18 15:04:20.723237908 +0800
+++ pastebinit/pastebinit	2014-01-18 15:04:22.219245346 +0800
@@ -65,7 +65,7 @@
     gettext.textdomain("pastebinit")
 
     # Timeout after 5s
-    socket.setdefaulttimeout(5)
+    socket.setdefaulttimeout(15)
 
     # Version number to show in the usage
     version = "1.4.1"
