=== modified file 'pastebinit'
--- orig/pastebinit	2014-01-17 00:38:17 +0000
+++ new/pastebinit	2014-01-18 04:51:35 +0000
@@ -1,5 +1,8 @@
 #!/usr/bin/python3
-# Written by Stephane Graber <stgraber@stgraber.org>
+# -*- coding: utf-8 -*-
+
+# Author: Stéphane Graber <stgraber@ubuntu.com>
+# Written by Stéphane Graber <stgraber@stgraber.org>
 #            Daniel Bartlett <dan@f-box.org>
 # Last modification : Mon Jan  6 18:46:46 EST 2014
 
@@ -65,7 +68,7 @@
     socket.setdefaulttimeout(5)
 
     # Version number to show in the usage
-    version = "1.4"
+    version = "1.4.1"
     configfile = os.path.expanduser("~/.pastebinit.xml")
 
     # Custom urlopener to handle 401's
@@ -99,7 +102,10 @@
 
                 filename = os.path.join(confdir, fileitem)
                 instance = SafeConfigParser()
-                instance.read(filename)
+                try:
+                    instance.read(filename)
+                except UnicodeError:
+                    continue
 
                 if not instance.has_section('pastebin'):
                     print(_('%s: no section [pastebin]') % filename,
@@ -187,7 +193,7 @@
         for node in nodelist:
             if node.nodeType == node.TEXT_NODE:
                 rc = rc + node.data
-        return rc.encode('utf-8')
+        return rc
 
     def getNodes(nodes, title):
         return nodes.getElementsByTagName(title)
@@ -339,9 +345,8 @@
             content = sys.stdin.read()
         else:
             try:
-                f = open(filename)
-                content = f.read()
-                f.close()
+                with open(filename, "rb") as fd:
+                    content = fd.read()
             except KeyboardInterrupt:
                 print(_("KeyboardInterrupt caught."), file=sys.stderr)
                 sys.exit(1)

