diff --git a/libraries/Makefile b/libraries/Makefile
index f5e279b..fa966f9 100644
--- a/libraries/Makefile
+++ b/libraries/Makefile
@@ -4,7 +4,7 @@ TARGET = libroboticscape.so
 SUBDIRS		:= $(shell find * -type d -print)
 CC		:= gcc
 LINKER		:= gcc
-CWARNINGS	:= -Wall -Wextra
+CWARNINGS	:= -Wall -Wextra -Wno-unused-result
 # achitecture flags enables the NEON hardware SIMD floating point unit
 ARCFLAGS	:= -mfpu=neon -mfloat-abi=hard -march=armv7-a -mtune=cortex-a8 
 # enable O3 optimization and vectorized math
