From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Tue, 9 Dec 2014 00:32:17 +0100
Subject: Add Debian specific SOVERSION

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b5bb44..392bd9f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,6 +29,7 @@ add_library(rospack
   src/rospack_cmdline.cpp
   src/utils.cpp
 )
+set_target_properties(rospack PROPERTIES VERSION ${rospack_VERSION} SOVERSION "0d")
 target_link_libraries(rospack ${TinyXML_LIBRARIES} ${Boost_LIBRARIES} ${PYTHON_LIBRARIES})
 
 add_executable(rospackexe src/rospack_main.cpp)
