From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Fri, 12 Dec 2014 07:48:34 +0100
Subject: Add usr/share as default ROS_PACKAGE_PATH

---
 src/rospack.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/rospack.cpp b/src/rospack.cpp
index efe6c11..1502a9e 100644
--- a/src/rospack.cpp
+++ b/src/rospack.cpp
@@ -257,6 +257,9 @@ bool
 Rosstackage::getSearchPathFromEnv(std::vector<std::string>& sp)
 {
   char* rpp = getenv("ROS_PACKAGE_PATH");
+
+  sp.push_back("/usr/share/");
+
   if(rpp)
   {
     // I can't see that boost filesystem has an elegant cross platform
