From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Mon, 9 Nov 2015 11:19:09 +0100
Subject: Make rospack usable without rosdep

---
 src/rospack.cpp | 19 ++++++++++++++++---
 1 file changed, 16 insertions(+), 3 deletions(-)

diff --git a/src/rospack.cpp b/src/rospack.cpp
index 1502a9e..d4c8fc2 100644
--- a/src/rospack.cpp
+++ b/src/rospack.cpp
@@ -1689,10 +1689,16 @@ Rosstackage::isSysPackage(const std::string& pkgname)
     Py_DECREF(pName);
     if(!pModule)
     {
-      PyErr_Print();
       PyGILState_Release(gstate);
       std::string errmsg = "could not find python module 'rosdep2.rospack'. is rosdep up-to-date (at least 0.10.4)?";
-      throw Exception(errmsg);
+
+      static bool rosdep_warning = false;
+      if(!rosdep_warning) {
+        std::cerr << "could not find python module 'rosdep2.rospack', continuing without it" << std::endl;
+        rosdep_warning = true;
+      }
+      cache[pkgname] = true;
+      return true;
     }
     pDict = PyModule_GetDict(pModule);
   }
@@ -1738,7 +1744,14 @@ Rosstackage::isSysPackage(const std::string& pkgname)
       PyErr_Print();
       PyGILState_Release(gstate);
       std::string errmsg = "the rosdep view is empty: call 'sudo rosdep init' and 'rosdep update'";
-      throw Exception(errmsg);
+
+      static bool rosdep_cache_warning = false;
+      if(!rosdep_cache_warning) {
+        std::cerr << "could not find rosdep cache, continuing without it" << std::endl;
+        rosdep_cache_warning = true;
+      }
+      cache[pkgname] = true;
+      return true;
     }
     rospack_view_not_empty = true;
   }
