diff --git a/host/mct-daemon/ti-mct-daemon.service.am57x b/host/mct-daemon/ti-mct-daemon.service.am57x
index 172eb32..d1ca16f 100644
--- a/host/mct-daemon/ti-mct-daemon.service.am57x
+++ b/host/mct-daemon/ti-mct-daemon.service.am57x
@@ -1,5 +1,7 @@
 [Unit]
 Description=TI MultiCore Tools Daemon
+ConditionPathExists=/sys/devices/platform/44000000.ocp/40800000.dsp/
+After=cmemk-module.service
 
 [Service]
 Type=forking
