From 2a49a27ed1d586f78c6a64b03241ba9f1e3b0dab Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 15 Oct 2019 13:57:49 -0500
Subject: [PATCH] bone101 dont install desktop icon

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 Makefile | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/Makefile b/Makefile
index b7e51c1..f78f3e0 100644
--- a/Makefile
+++ b/Makefile
@@ -29,10 +29,5 @@ install: ./bone101
 	cp -dr --preserve=mode,timestamp ./bone101/* $(DESTDIR)$(PREFIX)/share/bone101/
 	install -m 0775 -d $(DESTDIR)$(PREFIX)/share/applications
 	cp --preserve=mode,timestamp bone101.desktop $(DESTDIR)$(PREFIX)/share/applications/
-	install -m 0755 -d $(DESTDIR)/home/debian/Desktop
-	cp --preserve=mode,timestamp bone101.desktop $(DESTDIR)/home/debian/Desktop/
-	chown 1000:1000 $(DESTDIR)/home/debian
-	chown 1000:1000 $(DESTDIR)/home/debian/Desktop
-	chown 1000:1000 $(DESTDIR)/home/debian/Desktop/bone101.desktop
 
 .PHONY: clean test install
-- 
2.23.0

