From 69d8f2e7cb2925e6ba0c58101dc44953150ae54b Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 9 Jan 2018 13:56:49 -0600
Subject: [PATCH 2/2] try wifi

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 Support/script/beagle-ui.js | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Support/script/beagle-ui.js b/Support/script/beagle-ui.js
index 403d21c..a0eedf1 100644
--- a/Support/script/beagle-ui.js
+++ b/Support/script/beagle-ui.js
@@ -102,6 +102,7 @@ $(document).ready(function(){
             setTimeout(tryWindowHost, 5);
             setTimeout(try192, 5);
             setTimeout(try1926, 5);
+            setTimeout(trywifi, 5);
             setTimeout(tryLocal, 5);
 
             function tryWindowHost() {
@@ -113,6 +114,9 @@ $(document).ready(function(){
             function try1926() {
                 connectTarget('192.168.6.2', handlers);
             }
+            function trywifi() {
+                connectTarget('192.168.8.1', handlers);
+            }
             function tryLocal() {
                 connectTarget('beaglebone.local', handlers);
             }
-- 
2.15.1

