diff --git a/Makefile b/Makefile
index 43e0cb2..f37a24f 100644
--- a/Makefile
+++ b/Makefile
@@ -26,12 +26,11 @@ else
 endif
 
 install: ./bone101
-	install -m 0755 -d $(DESTDIR)$(PREFIX)/share/bone101
+	install -m 0775 -d $(DESTDIR)$(PREFIX)/share/bone101
 	cp -dr --preserve=mode,timestamp ./bone101/* $(DESTDIR)$(PREFIX)/share/bone101/
 	rm -rf $(DESTDIR)$(PREFIX)/share/bone101/debian/
-	install -m 0755 -d $(DESTDIR)$(PREFIX)/share/applications
+	install -m 0775 -d $(DESTDIR)$(PREFIX)/share/applications
 	cp --preserve=mode,timestamp bone101.desktop $(DESTDIR)$(PREFIX)/share/applications/
-	install -m 0755 -d $(DESTDIR)$(CLOUD9_DIR)
-	git clone --depth 1 $(EXAMPLES_REPO) $(DESTDIR)$(CLOUD9_DIR)$(CLOUD9_NAME)
+	install -m 0775 -d $(DESTDIR)$(CLOUD9_DIR)
 
 .PHONY: clean test install
