Description: add mips64el target host support to nodejs
Author: YunQiang Su <wzssyqa@gmail.com>
Last-Update: 2015-09-29
Forwarded: not yet
--- a/configure
+++ b/configure
@@ -27,7 +27,7 @@
 
 valid_os = ('win', 'mac', 'solaris', 'freebsd', 'openbsd', 'linux',
             'android', 'aix')
-valid_arch = ('arm', 'arm64', 'ia32', 'mips', 'mipsel', 'ppc', 'ppc64', 'x32',
+valid_arch = ('arm', 'arm64', 'ia32', 'mips', 'mipsel', 'mips64el', 'ppc', 'ppc64', 'x32',
               'x64', 'x86', 's390x')
 valid_arm_float_abi = ('soft', 'softfp', 'hard')
 valid_arm_fpu = ('vfp', 'vfpv3', 'vfpv3-d16', 'neon')
@@ -590,6 +590,9 @@
     '__s390x__'   : 's390x'
   }
 
+  if k.get('_MIPS_SIM') == '_ABI64' and k.get('__MIPSEL__') == '1':
+    return 'mips64el'
+
   rtn = 'ia32' # default
 
   for i in matchup:
@@ -674,7 +677,7 @@
 
   if target_arch == 'arm':
     configure_arm(o)
-  elif target_arch in ('mips', 'mipsel'):
+  elif target_arch in ('mips', 'mipsel', 'mips64el'):
     configure_mips(o)
 
   if flavor == 'aix':
