From ba4533578906d6ca96dfeaf3dd8037714855e79e Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 19 Jan 2017 10:48:22 -0600
Subject: [PATCH] Makefile: use /usr/bin/dtc

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 857f2b8..7d1cb73 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 
-DTC ?= /usr/bin/dtc-v4.1.x
+DTC ?= /usr/bin/dtc
 #DTC ?= /usr/bin/dtc-v1.4.2-git
 CPP ?= cpp
 DESTDIR ?=
-- 
2.11.0

