Description: Some typos fixed in manpages
Author: Alf Gaida <agaida@siduction.org>
Last-Update: 2018-02-17

--- connman-1.35.orig/doc/connman-vpn-provider.config.5.in
+++ connman-1.35/doc/connman-vpn-provider.config.5.in
@@ -12,7 +12,7 @@ connection_name.config \- ConnMan vpn co
 \fIConnMan\fP's vpn connections are configured with so called
 "\fBprovisioning files\fP" which reside under \fI@vpn_storagedir@/\fP.
 The files can be named anything, as long as they contain only printable
-ascii characers, for example letters, numbers and underscores. The file
+ascii characters, for example letters, numbers and underscores. The file
 must end with \fB.config\fP. Each VPN connection requires a provisioning
 file, but multiple connections can be specified in the same file.
 .SH "FILE FORMAT"
--- connman-1.35.orig/doc/connman.conf.5.in
+++ connman-1.35/doc/connman.conf.5.in
@@ -71,7 +71,7 @@ set up and saved to storage beforehand.
 List of technoolgies which are always connected regardless
 of PreferredTechnologies setting (AutoConnect = true). The
 default value is empty and this feature is disabled unless
-explicitely enabled in the config file.
+explicitly enabled in the config file.
 .TP
 .BI PreferredTechnologies= technology\fR[,...]
 List of preferred technologies from the most preferred
