From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Wed, 5 Aug 2015 08:44:03 +0200
Subject: Add Debian specific SOVERSION

---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9ae6b09..da0181f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,6 +65,7 @@ add_library(${PROJECT_NAME}
   src/shape_to_marker.cpp
   src/shape_extents.cpp)
 target_link_libraries(${PROJECT_NAME} ${ASSIMP_LIBRARIES} ${QHULL_LIBRARIES} ${catkin_LIBRARIES} ${console_bridge_LIBRARIES} ${Boost_LIBRARIES})
+set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${geometric_shapes_VERSION} SOVERSION "0d")
 
 
 if(CATKIN_ENABLE_TESTING)
@@ -78,4 +79,4 @@ install(TARGETS ${PROJECT_NAME}
 
 install(DIRECTORY include/
         DESTINATION include
-        FILES_MATCHING PATTERN "*.h")
\ No newline at end of file
+        FILES_MATCHING PATTERN "*.h")
