Description: update default/supported Python versions in fallback code
Author: Piotr Ozarowski <piotr@debian.org>

Index: dh-python/dhpython/_defaults.py
===================================================================
--- dh-python.orig/dhpython/_defaults.py
+++ dh-python/dhpython/_defaults.py
@@ -26,12 +26,12 @@ from os.path import exists
 from subprocess import Popen, PIPE
 
 SUPPORTED = {
-    'cpython2': [(2, 7)],
-    'cpython3': [(3, 4)],
+    'cpython2': [(2, 7), (2, 6)],
+    'cpython3': [(3, 2)],
     'pypy': [(2, 0)]}
 DEFAULT = {
     'cpython2': (2, 7),
-    'cpython3': (3, 4),
+    'cpython3': (3, 2),
     'pypy': (2, 0)}
 
 log = logging.getLogger('dhpython')
