From 1c04d5e78040b5bcd7d11e43e965edf372b03449 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 4 Apr 2017 13:15:50 -0500
Subject: [PATCH] drop toolchain

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 Tools/ardupilotwaf/boards.py | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/Tools/ardupilotwaf/boards.py b/Tools/ardupilotwaf/boards.py
index 26e73b7ed..e49e0bc7e 100644
--- a/Tools/ardupilotwaf/boards.py
+++ b/Tools/ardupilotwaf/boards.py
@@ -296,8 +296,6 @@ class zynq(linux):
         )
 
 class bbbmini(linux):
-    toolchain = 'arm-linux-gnueabihf'
-
     def configure_env(self, cfg, env):
         super(bbbmini, self).configure_env(cfg, env)
 
@@ -306,8 +304,6 @@ class bbbmini(linux):
         )
 
 class blue(linux):
-    toolchain = 'arm-linux-gnueabihf'
-
     def configure_env(self, cfg, env):
         super(blue, self).configure_env(cfg, env)
 
-- 
2.11.0

