From: Zoltan Gyarmati <mr.zoltan.gyarmati@gmail.com>
Date: Mon, 27 Feb 2017 00:17:34 +0100
Subject: check explicity for __GLIBC__ to set the feature macros also in
 kfreebsd, closes: #816334

---
 libserialport_internal.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libserialport_internal.h b/libserialport_internal.h
index 669152b..69d8113 100644
--- a/libserialport_internal.h
+++ b/libserialport_internal.h
@@ -21,8 +21,8 @@
 #ifndef LIBSERIALPORT_LIBSERIALPORT_INTERNAL_H
 #define LIBSERIALPORT_LIBSERIALPORT_INTERNAL_H
 
-
-#ifdef __linux__
+/* We need to check also for __GLIBC__ for kfreebsd */
+#if defined( __GLIBC__) || defined(__linux__)
 /* For timeradd, timersub, timercmp. */
 #define _BSD_SOURCE 1 /* for glibc < 2.19 */
 #define _DEFAULT_SOURCE 1 /* for glibc >= 2.20 */
