Index: openssl-1.0.0c/crypto/rand/md_rand.c
===================================================================
--- openssl-1.0.0c.orig/crypto/rand/md_rand.c	2010-06-16 15:17:22.000000000 +0200
+++ openssl-1.0.0c/crypto/rand/md_rand.c	2010-12-12 17:02:50.000000000 +0100
@@ -476,6 +476,7 @@
 		MD_Update(&m,(unsigned char *)&(md_c[0]),sizeof(md_c));
 
 #ifndef PURIFY /* purify complains */
+#if 0
 		/* The following line uses the supplied buffer as a small
 		 * source of entropy: since this buffer is often uninitialised
 		 * it may cause programs such as purify or valgrind to
@@ -485,6 +486,7 @@
 		 */
 		MD_Update(&m,buf,j);
 #endif
+#endif
 
 		k=(st_idx+MD_DIGEST_LENGTH/2)-st_num;
 		if (k > 0)
