From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Tue, 4 Aug 2015 20:13:40 +0200
Subject: Add CMakeLists.txt

---
 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..ac2897b
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(vision_opencv)
+
+add_subdirectory(cv_bridge)
+add_subdirectory(image_geometry)
+add_subdirectory(opencv_apps)
+add_subdirectory(opencv_tests)
