--- a/misc/desktop/cmst-autostart.desktop
+++ b/misc/desktop/cmst-autostart.desktop
@@ -1,13 +1,14 @@
 [Desktop Entry]
 Type=Application
 Version=1.1
-Terminal=false
-StartupNotify=false
-Name=CMST
+Name=Connman Setup (Autostart)
 GenericName=Network Configuration
 Comment=QT GUI frontend for connman
-Categories=Network;Utility
-Icon=/usr/share/icons/hicolor/24x24/apps/cmst.png
+Categories=Settings;System;Qt;NetworkSettings
+Icon=cmst
 Terminal=false
-Exec=/usr/bin/cmst
+Exec=cmst
+StartupNotify=false
 X-GNOME-Autostart-enabled=true
+
+Name[de]=Connman Einstellungen (Autostart)
\ No newline at end of file
--- a/misc/desktop/cmst.desktop
+++ b/misc/desktop/cmst.desktop
@@ -1,10 +1,14 @@
 [Desktop Entry]
 Type=Application
 Version=1.1
-Name=CMST
+Name=Connman Setup
 GenericName=Network Configuration
 Comment=QT GUI frontend for connman
-Categories=Network;Utility
-Icon=/usr/share/icons/hicolor/24x24/apps/cmst.png
+Categories=Settings;System;Qt;NetworkSettings
+Icon=cmst
 Terminal=false
-Exec=/usr/bin/cmst
+Exec=cmst
+StartupNotify=false
+X-GNOME-Autostart-enabled=true
+
+Name[de]=Connman Einstellungen
\ No newline at end of file
