Description: Categories and OnlyShowIn fixed
Author: Alf Gaida <agaida@siduction.org>

---
Last-Update: 2014-10-17

--- lxqt-about-0.8.0.orig/resources/lxqt-about.desktop.in
+++ lxqt-about-0.8.0/resources/lxqt-about.desktop.in
@@ -5,7 +5,7 @@ GenericName=About LXQt
 Comment=Learn more about LXQt
 Exec=lxqt-about
 Icon=help-about
-Categories=Help;LXQt
-OnlyShowIn=LXQt;LXDE;
+Categories=Settings;Documentation;Qt;
+OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
