From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Mon, 30 Nov 2015 18:45:18 +0100
Subject: Add default plugin path

---
 include/pluginlib/class_loader_imp.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/pluginlib/class_loader_imp.h b/include/pluginlib/class_loader_imp.h
index 9d9543c..38f1db5 100644
--- a/include/pluginlib/class_loader_imp.h
+++ b/include/pluginlib/class_loader_imp.h
@@ -247,6 +247,7 @@ namespace pluginlib
     std::vector<std::string> all_paths;
     std::vector<std::string> all_paths_without_extension = getCatkinLibraryPaths();
     all_paths_without_extension.push_back(getROSBuildLibraryPath(exporting_package_name));
+    all_paths_without_extension.push_back("/usr/lib/" + exporting_package_name);
     bool debug_library_suffix = (class_loader::systemLibrarySuffix().compare(0, 1, "d") == 0);
     std::string non_debug_suffix;
     if(debug_library_suffix) {
