From f2c1798d100b4c09ee461b63019ff6d291c1b674 Mon Sep 17 00:00:00 2001
From: Jason Kridner <jkridner@beagleboard.org>
Date: Thu, 30 Jun 2016 20:18:10 +0000
Subject: [PATCH 1/4] Makefile: build before install

---
 Makefile | 25 +++++++++++--------------
 1 file changed, 11 insertions(+), 14 deletions(-)

diff --git a/Makefile b/Makefile
index d315db9..6e890f5 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 PREFIX:=/usr/local
-npm_options:=--unsafe-perm=true --progress=false --loglevel=error --prefix $(DESTDIR)$(PREFIX)
+npm_options:=--unsafe-perm=true --progress=false --loglevel=error
 NODE_PATH:=$(DESTDIR)$(PREFIX)/lib/node_modules
 NODEJS:=nodejs
 NPM:=npm
@@ -7,29 +7,28 @@ NPM:=npm
 ver=$(shell nodejs -pe "require('./package.json').version")
 npm_version=$(shell $(NPM) --version)
 node_version=$(shell $(NODEJS) --version)
+bs_path=./build/lib/node_modules/bonescript
 
 all:
-	@echo "ver=$(ver)"
-	@echo "npm_version=$(npm_version)"
-	@echo "node_version=$(node_version)"
+	mkdir -p $(bs_path)
+	echo $(ver) > $(bs_path)/bonescript.version
+	echo $(node_version) > $(bs_path)/bonescript.node_version
+	echo $(npm_version) > $(bs_path)/bonescript.npm_version
+	TERM=dumb $(NPM) install -g $(npm_options) --prefix ./build
+	cp -dr --preserve=mode,timestamp etc/* ./build/etc/
 
 clean:
-
+	rm -rf build
 
 test:
 
 
 install:
-	NODE_PATH=$(NODE_PATH) TERM=dumb $(NPM) install -g $(npm_options)
-	echo $(ver) > $(NODE_PATH)/bonescript/bonescript.version
-	echo $(node_version) > $(NODE_PATH)/bonescript/bonescript.node_version
+	install -m 0755 -d $(DESTDIR)$(PREFIX)
+	cp -dr --preserve=mode,timestamp ./build/* $(DESTDIR)$(PREFIX)/
 	systemctl stop bonescript-autorun.service || true
 	systemctl stop bonescript.service || true
 	systemctl stop bonescript.socket || true
-	install -m 0755 -d $(DESTDIR)/etc/default
-	install -m 0644 etc/default/node $(DESTDIR)/etc/default/
-	install -m 0755 -d $(DESTDIR)/etc/profile.d
-	install -m 0755 etc/profile.d/node.sh $(DESTDIR)/etc/profile.d/
 	install -m 0755 -d $(DESTDIR)/lib/systemd/system
 	install -m 0644 systemd/bonescript.socket $(DESTDIR)/lib/systemd/system/
 	install -m 0644 systemd/bonescript.service $(DESTDIR)/lib/systemd/system/
@@ -37,7 +36,5 @@ install:
 	systemctl enable bonescript.socket || true
 	systemctl enable bonescript.service || true
 	systemctl enable bonescript-autorun.service || true
-	install -m 0755 -d $(DESTDIR)/etc/avahi/services
-	install -m 0644 etc/avahi/services/bone101.service $(DESTDIR)/etc/avahi/services/
 
 .PHONY: clean test install
-- 
2.8.1

