From: Leopold Palomo-Avellaneda <leopold.palomo@upc.edu>
Date: Thu, 30 Apr 2015 08:44:24 +0200
Subject: Added CMakeLists to build thepackage

---
 CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..72f44fa
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,6 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(geometry)
+add_subdirectory( eigen_conversions )
+add_subdirectory( kdl_conversions )
+add_subdirectory( tf )
+add_subdirectory( tf_conversions )
