From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Wed, 5 Aug 2015 14:05:24 +0200
Subject: Add Debian specific SOVERSION

---
 src/rviz/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/rviz/CMakeLists.txt b/src/rviz/CMakeLists.txt
index 5dad564..40a87cd 100644
--- a/src/rviz/CMakeLists.txt
+++ b/src/rviz/CMakeLists.txt
@@ -198,6 +198,7 @@ target_link_libraries(${PROJECT_NAME}
   assimp
   yaml-cpp
 )
+set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${rviz_VERSION} SOVERSION "0d")
 
 if(APPLE)
   set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
