From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Sat, 5 Dec 2015 18:52:09 +0100
Subject: Remove default_plugin from exported libraries

---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 401540e..8a4a5ae 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -143,7 +143,6 @@ catkin_package(
     ${OPENGL_INCLUDE_DIR}
   LIBRARIES
     rviz
-    default_plugin
     ${OGRE_OV_LIBRARIES_ABS}
     ${rviz_ADDITIONAL_LIBRARIES}
     ${OPENGL_LIBRARIES}
