From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Tue, 17 Nov 2015 00:01:41 +0100
Subject: Add default ROS_ROOT to rosmake

---
 tools/rosmake/src/rosmake/engine.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tools/rosmake/src/rosmake/engine.py b/tools/rosmake/src/rosmake/engine.py
index 4d07047..5307101 100755
--- a/tools/rosmake/src/rosmake/engine.py
+++ b/tools/rosmake/src/rosmake/engine.py
@@ -381,6 +381,8 @@ class RosMakeAll:
             with _popen_lock:
                 num_cpus = parallel_build.num_cpus()
                 local_env['ROS_PARALLEL_JOBS'] = "-j%d -l%d" % (num_cpus, num_cpus)
+        if 'ROS_ROOT' not in os.environ:
+                local_env['ROS_ROOT'] = '/usr/share/ros'
         local_env['SVN_CMDLINE'] = "svn --non-interactive"
         cmd = ["bash", "-c", "cd %s && %s "%(self.rospack.get_path(package), make_command()) ] #UNIXONLY
         if argument:
