diff --git a/battery_monitor_service/rc_battery_monitor/rc_battery_monitor.c b/battery_monitor_service/rc_battery_monitor/rc_battery_monitor.c
index 736bf94..d5d40d5 100644
--- a/battery_monitor_service/rc_battery_monitor/rc_battery_monitor.c
+++ b/battery_monitor_service/rc_battery_monitor/rc_battery_monitor.c
@@ -80,9 +80,10 @@ int main(int argc, char *argv[]){
     // whitelist blue, black, and black wireless only when RC device tree is in use
     model = rc_get_bb_model();
     if(model!=BB_BLACK_RC && model!=BB_BLACK_W_RC && model!=BB_BLUE){
-    	if(system("grep -q roboticscape /boot/uEnv.txt")!=0){
+    	if(system("grep -q roboticscape /boot/uEnv.txt | grep -v '#'")!=0){
 			fprintf(stderr,"rc_battery_monitor can only run on BB Blue, Black, and Black wireless when the roboticscape device tree is in use.\n");
-			return -1;
+			kill_existing_instance();
+			return 0;
 		}
     }
 
diff --git a/roboticscape_service/rc_startup_routine/rc_startup_routine.c b/roboticscape_service/rc_startup_routine/rc_startup_routine.c
index 1c8dae4..bc3f432 100644
--- a/roboticscape_service/rc_startup_routine/rc_startup_routine.c
+++ b/roboticscape_service/rc_startup_routine/rc_startup_routine.c
@@ -48,9 +48,9 @@ int main(){
 	// whitelist blue, black, and black wireless only when RC device tree is in use
     model = rc_get_bb_model();
     if(model!=BB_BLACK_RC && model!=BB_BLACK_W_RC && model!=BB_BLUE){
-    	if(system("grep -q roboticscape /boot/uEnv.txt")!=0){
+    	if(system("grep -q roboticscape /boot/uEnv.txt | grep -v '#'")!=0){
 			fprintf(stderr,"roboticscape service can only run on BB Blue, Black, and Black wireless when the roboticscape device tree is in use.\n");
-			return -1;
+			return 0;
 		}
     }
 
