From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Tue, 2 Jun 2015 10:28:32 +0200
Subject: Add Debian specific SOVERSION

---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ee9a601..e0aa385 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,6 +34,7 @@ set(${PROJECT_NAME}_SRCS
 )
 add_library(${PROJECT_NAME} ${${PROJECT_NAME}_SRCS})
 target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES} ${console_bridge_LIBRARIES} ${Poco_LIBRARIES})
+set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${class_loader_VERSION} SOVERSION "0d")
 
 install(TARGETS ${PROJECT_NAME}
   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
@@ -60,4 +61,4 @@ if(NOT ${catkin_FOUND})
                  ${CMAKE_CURRENT_BINARY_DIR}/${TARGET_NAME}.pc @ONLY)
 
   install(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION lib/pkgconfig)
-endif()
\ No newline at end of file
+endif()
