From ba41ded30e004f466412973bea84fdc60f61869f Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 6 Oct 2015 16:04:48 -0500
Subject: [PATCH] scripts/functions: add PARTUUID to resolve_device function

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 scripts/functions | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/scripts/functions b/scripts/functions
index 8c1bb1f..4877ae2 100644
--- a/scripts/functions
+++ b/scripts/functions
@@ -321,6 +321,9 @@ resolve_device() {
 		esac
 		DEV="/dev/disk/by-label/${DEV}"
 		;;
+	PARTUUID=*)
+		DEV="/dev/disk/by-partuuid/${DEV#PARTUUID=}"
+		;;
 	UUID=*)
 		DEV="/dev/disk/by-uuid/${DEV#UUID=}"
 		;;
-- 
2.5.3

