Description: Fix OnlyShowIn=LXQt;
Author: Alf Gaida <agaida@siduction.org>
Last-Update: 2014-10-19

--- lxqt-powermanagement-0.8.0.orig/config/lxqt-config-powermanagement.desktop.in
+++ lxqt-powermanagement-0.8.0/config/lxqt-config-powermanagement.desktop.in
@@ -5,6 +5,6 @@ GenericName=Power Management Settings
 Exec=lxqt-config-powermanagement
 Icon=preferences-system-power-management
 Categories=Settings;DesktopSettings;Qt;LXQt;
-OnlyShowIn=LXQt;LXDE;
+OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
