commit 189017c078b7870c18ced485c1fd99f65fcc4801
Author: nicm <nicm>
Date:   Thu May 8 06:06:07 2014 +0000

    Plug a memory leak, from J Raynor.

--- a/cmd-find-window.c
+++ b/cmd-find-window.c
@@ -199,6 +199,8 @@
 	window_choose_ready(wl->window->active, 0, cmd_find_window_callback);
 
 out:
+	for (i = 0; i < ARRAY_LENGTH(&find_list); i++)
+		free(ARRAY_ITEM(&find_list, i).list_ctx);
 	ARRAY_FREE(&find_list);
 	return (CMD_RETURN_NORMAL);
 }
