diff --git a/configure.ac b/configure.ac
index d215b63..a08870b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -553,6 +553,14 @@ AC_LINK_IFELSE(
         GC_SECTIONS="";])
 LDFLAGS=$save_LDFLAGS
 
+# XXX workaround for Debian Bug #844357
+# --gc-sections is currently broken on mips* with binutils > 2.27-9
+# Debian bug: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=844357
+# Upstream bug: https://sourceware.org/bugzilla/show_bug.cgi?id=20828
+case "$host_cpu" in
+    mips*) GC_SECTIONS="" ;;
+esac
+
 AC_SUBST([GC_SECTIONS])
 
 dnl
diff --git a/src/gallium/targets/va/Makefile.am b/src/gallium/targets/va/Makefile.am
index df825b7..12724a1 100644
--- a/src/gallium/targets/va/Makefile.am
+++ b/src/gallium/targets/va/Makefile.am
@@ -29,6 +29,7 @@ gallium_drv_video_la_LIBADD = \
 	$(top_builddir)/src/gallium/auxiliary/libgalliumvl.la \
 	$(top_builddir)/src/gallium/auxiliary/libgallium.la \
 	$(top_builddir)/src/util/libmesautil.la \
+	$(top_builddir)/src/compiler/nir/libnir.la \
 	$(VL_LIBS) \
 	$(LIBDRM_LIBS) \
 	$(GALLIUM_COMMON_LIB_DEPS)
diff --git a/src/gallium/targets/vdpau/Makefile.am b/src/gallium/targets/vdpau/Makefile.am
index d388f8b..6f0f9e5 100644
--- a/src/gallium/targets/vdpau/Makefile.am
+++ b/src/gallium/targets/vdpau/Makefile.am
@@ -34,6 +34,7 @@ libvdpau_gallium_la_LIBADD = \
 	$(top_builddir)/src/gallium/auxiliary/libgalliumvl.la \
 	$(top_builddir)/src/gallium/auxiliary/libgallium.la \
 	$(top_builddir)/src/util/libmesautil.la \
+	$(top_builddir)/src/compiler/nir/libnir.la \
 	$(VL_LIBS) \
 	$(LIBDRM_LIBS) \
 	$(GALLIUM_COMMON_LIB_DEPS)
