diff --git a/Makefile.am b/Makefile.am
index 5272bdb..35805c0 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -35,6 +35,12 @@
 SUBDIRS =  linux/etc linux/src/utils linux/src/api linux/src/transport \
         linux/src/mm linux/src/daemon linux/src/tests
 
+# hook to remove libtools library extention files (.la)
+install-exec-hook:
+	rm -f $(DESTDIR)$(libdir)/*.la
+
+ACLOCAL_AMFLAGS = -I m4
+
 # where to install common headers on the system
 ipcincludedir = $(includedir)/ti/ipc
 ipcincludeinterfacesdir = $(includedir)/ti/ipc/interfaces
diff --git a/Makefile.in b/Makefile.in
index 1ecdf33..6cacf1e 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -867,6 +867,8 @@ install-dvi: install-dvi-recursive
 install-dvi-am:
 
 install-exec-am:
+	@$(NORMAL_INSTALL)
+	$(MAKE) $(AM_MAKEFLAGS) install-exec-hook
 
 install-html: install-html-recursive
 
@@ -935,6 +937,10 @@ uninstall-am: uninstall-ipcincludeHEADERS \
 .PRECIOUS: Makefile
 
 
+
+# hook to remove libtools library extention files (.la)
+install-exec-hook:
+	rm -f $(DESTDIR)$(libdir)/*.la
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff --git a/configure.ac b/configure.ac
index 4d4a559..3776de9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -43,6 +43,9 @@ AC_CONFIG_SRCDIR([packages/ti/ipc/MessageQ.h])
 # Set auxiliary build tools directory
 AC_CONFIG_AUX_DIR([linux/build-aux])
 
+AC_SEARCH_LIBS([pthread_create], [pthread])
+AC_CONFIG_MACRO_DIR([m4])
+
 # Initilize automake
 AM_INIT_AUTOMAKE([foreign])
 
diff --git a/m4/.gitignore b/m4/.gitignore
new file mode 100644
index 0000000..38066dd
--- /dev/null
+++ b/m4/.gitignore
@@ -0,0 +1,5 @@
+libtool.m4
+ltoptions.m4
+ltsugar.m4
+ltversion.m4
+lt~obsolete.m4
-- 
2.6.2

