From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Wed, 5 Aug 2015 13:33:39 +0200
Subject: Add CMakeLists.txt

---
 CMakeLists.txt | 9 +++++++++
 1 file changed, 9 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..4b3842c
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(robot_model)
+
+add_subdirectory(urdf_parser_plugin)
+add_subdirectory(urdf)
+add_subdirectory(collada_parser)
+add_subdirectory(collada_urdf)
+add_subdirectory(joint_state_publisher)
+add_subdirectory(kdl_parser)
