diff --git a/package.json b/package.json
index 0e40b46..ff0bf09 100644
--- a/package.json
+++ b/package.json
@@ -53,6 +53,7 @@
     "node-red-node-rbe": "^0.2.9",
     "node-red-node-tail": "^0.1.0",
     "nopt": "4.0.3",
+    "systemd": "0.3.1",
     "semver": "6.3.0"
   },
   "deprecated": false,
diff --git a/red.js b/red.js
index 32639ea..fd2e0e8 100755
--- a/red.js
+++ b/red.js
@@ -38,6 +38,8 @@ var path = require("path");
 var fs = require("fs-extra");
 var RED = require("./lib/red.js");
 
+require('systemd');
+
 var server;
 var app = express();
 
@@ -281,14 +283,15 @@ httpsPromise.then(function(startupHttps) {
 
     // if we got a port from command line, use it (even if 0)
     // replicate (settings.uiPort = parsedArgs.port||settings.uiPort||1880;) but allow zero
-    if (parsedArgs.port !== undefined){
-        settings.uiPort = parsedArgs.port;
-    } else {
-        if (settings.uiPort === undefined){
-            settings.uiPort = 1880;
-        }
-    }
-
+    //if (parsedArgs.port !== undefined){
+    //    settings.uiPort = parsedArgs.port;
+    //} else {
+    //    if (settings.uiPort === undefined){
+    //        settings.uiPort = 1880;
+    //    }
+    //}
+
+    settings.uiPort = parsedArgs.port||settings.uiPort||1880;
     settings.uiHost = settings.uiHost||"0.0.0.0";
 
     if (flowFile) {
@@ -380,14 +383,14 @@ httpsPromise.then(function(startupHttps) {
     }
 
     function getListenPath() {
-        var port = settings.serverPort;
-        if (port === undefined){
-            port = settings.uiPort;
-        }
+        //var port = settings.serverPort;
+        //if (port === undefined){
+        //    port = settings.uiPort;
+        //}
 
         var listenPath = 'http'+(settings.https?'s':'')+'://'+
                         (settings.uiHost == '::'?'localhost':(settings.uiHost == '0.0.0.0'?'127.0.0.1':settings.uiHost))+
-                        ':'+port;
+                        ':'+settings.uiPort;
         if (settings.httpAdminRoot !== false) {
             listenPath += settings.httpAdminRoot;
         } else if (settings.httpStatic) {
@@ -418,11 +421,13 @@ httpsPromise.then(function(startupHttps) {
                 RED.log[delayedLogItem.type](RED.log._(delayedLogItem.id, delayedLogItem.params||{}));
             });
 
-            server.listen(settings.uiPort,settings.uiHost,function() {
+            //server.listen(settings.uiPort,settings.uiHost,function() {
+            server.listen(settings.uiPort);
+            server.on('listening', function() {
                 if (settings.httpAdminRoot === false) {
                     RED.log.info(RED.log._("server.admin-ui-disabled"));
                 }
-                settings.serverPort = server.address().port;
+                //settings.serverPort = server.address().port;
                 process.title = parsedArgs.title || 'node-red';
                 RED.log.info(RED.log._("server.now-running", {listenpath:getListenPath()}));
             });
diff --git a/settings.js b/settings.js
index f3c93f5..8b68ad3 100644
--- a/settings.js
+++ b/settings.js
@@ -14,9 +14,12 @@
  * limitations under the License.
  **/
 
+var port = process.env.LISTEN_PID > 0 ? 'systemd' : 1880;
+
 module.exports = {
     // the tcp port that the Node-RED web server is listening on
-    uiPort: process.env.PORT || 1880,
+    //uiPort: process.env.PORT || 1880,
+    uiPort: process.env.PORT || port,
 
     // By default, the Node-RED UI accepts connections on all IPv4 interfaces.
     // To listen on all IPv6 addresses, set uiHost to "::",
