diff --git a/config/default.js b/config/default.js
index 89061be..1da804c 100644
--- a/config/default.js
+++ b/config/default.js
@@ -14,7 +14,7 @@ const home = os.homedir();
 
 module.exports = {
   // Base profile directory
-  profileDir: `${home}/.webthings`,
+  profileDir: '/etc/webthings-gateway',
 
   ports: {
     // HTTPS port
@@ -29,7 +29,7 @@ module.exports = {
 
   // Whether the gateway is behind port forwarding and should use simplified
   // port-free urls
-  behindForwarding: true,
+  behindForwarding: false,
 
   addonManager: {
     // URLs of add-on lists to parse, in order
@@ -69,7 +69,7 @@ module.exports = {
     domain: 'webthings.io',
 
     // Command to run PageKite
-    pagekite_cmd: path.normalize(path.join(process.cwd(), 'pagekite.py')),
+    pagekite_cmd: '/usr/bin/pagekite',
 
     // Port the PageKite server is running on
     port: 443,
-- 
2.30.1

