description: support for more gpu devices
origin: https://github.com/llvm-mirror/libclc/commit/1b1b5a8551971ef4062bd8d84baccfe7a88e83b2

--- a/configure.py
+++ b/configure.py
@@ -103,7 +103,7 @@ available_targets = {
                 {'gpu' : 'barts',   'aliases' : ['turks', 'caicos']},
                 {'gpu' : 'cayman',  'aliases' : ['aruba']}]},
   'amdgcn--': { 'devices' :
-                [{'gpu' : 'tahiti',  'aliases' : ['pitcairn', 'verde', 'oland', 'hainan', 'bonaire', 'kabini', 'kaveri', 'hawaii','mullins']}]},
+                [{'gpu' : 'tahiti',  'aliases' : ['pitcairn', 'verde', 'oland', 'hainan', 'bonaire', 'kabini', 'kaveri', 'hawaii', 'mullins', 'tonga', 'carrizo', 'iceland', 'fiji', 'stoney']}]},
   'nvptx--'   : { 'devices' : [{'gpu' : '', 'aliases' : []}]},
   'nvptx64--'   : { 'devices' : [{'gpu' : '', 'aliases' : []}] },
   'nvptx--nvidiacl'   : { 'devices' : [{'gpu' : '', 'aliases' : []}] },
