diff --git a/Makefile b/Makefile
index 9fc8e0d..f7352b0 100644
--- a/Makefile
+++ b/Makefile
@@ -14,6 +14,12 @@ else
 	$(error jekyll not installed)
 endif
 ifdef NPM
+	mkdir -p $(CURDIR)/debian/tmp/.npm
+	mkdir -p $(CURDIR)/debian/tmp/.npmrc
+	export HOME=$(CURDIR)
+	$(NPM) npm config ls -l
+	$(NPM) config set cache $(CURDIR)/debian/tmp/.npm
+	$(NPM) config set userconfig $(CURDIR)/debian/tmp/.npmrc
 	mkdir -p node_modules
 	cd node_modules
 	$(NPM) install --unsafe-perm=true --progress=false --loglevel=error async@2.0.0-rc.6
