diff --git a/Makefile b/Makefile
index 9fc8e0d..73c7fd5 100644
--- a/Makefile
+++ b/Makefile
@@ -14,6 +14,8 @@ else
 	$(error jekyll not installed)
 endif
 ifdef NPM
+	$(NPM) config set cache $(CURDIR)/debian/tmp/.npm
+	$(NPM) config set userconfig $(CURDIR)/debian/tmp/.npmrc
 	mkdir -p node_modules
 	cd node_modules
 	$(NPM) install --unsafe-perm=true --progress=false --loglevel=error async@2.0.0-rc.6
