diff --git a/Makefile b/Makefile
index 9fc8e0d..b083758 100644
--- a/Makefile
+++ b/Makefile
@@ -16,8 +16,8 @@ endif
 ifdef NPM
 	mkdir -p node_modules
 	cd node_modules
-	$(NPM) install --unsafe-perm=true --progress=false --loglevel=error async@2.0.0-rc.6
-	$(NPM) install --unsafe-perm=true --progress=false --loglevel=error sensortag@1.2.2
+	tar xf ../debian/async-2.0.0-rc.6-v0.12.15.tar.xz -C ./
+	tar xf ../debian/sensortag-1.2.2-v0.12.15.tar.xz -C ./
 	cd ..
 else
 	@echo npm not found
