Description: Adapt paths to system-wide install
Forwarded: not-needed
Author: Marcelo Jorge Vieira <metal@alucinados.com>
Author: Jonas Smedegaard <dr@jones.dk>
Last-Update: 2014-05-02

--- a/bin/uglifyjs
+++ b/bin/uglifyjs
@@ -3,7 +3,7 @@
 
 "use strict";
 
-var UglifyJS = require("../tools/node");
+var UglifyJS = require("uglify-js");
 var sys = require("util");
 var optimist = require("optimist");
 var fs = require("fs");
--- a/test/run-tests.js
+++ b/test/run-tests.js
@@ -1,6 +1,6 @@
 #! /usr/bin/env node
 
-var U = require("../tools/node");
+var U = require("uglify-js");
 var path = require("path");
 var fs = require("fs");
 var assert = require("assert");
--- a/tools/node.js
+++ b/tools/node.js
@@ -28,15 +28,15 @@
 };
 
 var FILES = exports.FILES = [
-    "../lib/utils.js",
-    "../lib/ast.js",
-    "../lib/parse.js",
-    "../lib/transform.js",
-    "../lib/scope.js",
-    "../lib/output.js",
-    "../lib/compress.js",
-    "../lib/sourcemap.js",
-    "../lib/mozilla-ast.js"
+    "./utils.js",
+    "./ast.js",
+    "./parse.js",
+    "./transform.js",
+    "./scope.js",
+    "./output.js",
+    "./compress.js",
+    "./sourcemap.js",
+    "./mozilla-ast.js"
 ].map(function(file){
     return path.join(path.dirname(fs.realpathSync(__filename)), file);
 });
