From 77b3ed07a7a0ff0128fc60303d98637a8dce158e Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 26 Apr 2016 14:42:48 -0500
Subject: [PATCH] seeed: defaults

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 wifidog.conf | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/wifidog.conf b/wifidog.conf
index fd953f0..cb5b080 100644
--- a/wifidog.conf
+++ b/wifidog.conf
@@ -33,7 +33,7 @@
 # and eth1, wlan0, ath0, etc. otherwise
 # You can get this interface with the ifconfig command and finding your wifi interface
 
-GatewayInterface br-lan
+GatewayInterface SoftAp0
 
 # Parameter: GatewayAddress
 # Default: Find it from GatewayInterface
@@ -41,7 +41,7 @@ GatewayInterface br-lan
 #
 # Set this to the internal IP address of the gateway.  Not normally required.
 
-# GatewayAddress 192.168.1.1
+GatewayAddress 192.168.8.1
 
 # Parameter: HtmlMessageFile
 # Default: wifidog-msg.html
@@ -90,6 +90,13 @@ GatewayInterface br-lan
 #    Path /
 #}
 
+AuthServer {
+    Hostname 192.168.8.1
+    SSLAvailable no
+    HTTPPort 520
+    Path /
+}
+
 # Parameter: DeltaTraffic
 # Default: no
 # Optional
-- 
2.8.0.rc3

