Description: Adapt paths to system-wide install
Forwarded: not-needed
Author: Marcelo Jorge Vieira <metal@alucinados.com>
Author: Jonas Smedegaard <dr@jones.dk>
Last-Update: 2012-12-06

--- a/uglify-js.js
+++ b/uglify-js.js
@@ -11,8 +11,8 @@
   return final_code;
 };
 
-uglify.parser = require("./lib/parse-js");
-uglify.uglify = require("./lib/process");
-uglify.consolidator = require("./lib/consolidator");
+uglify.parser = require("./parse-js");
+uglify.uglify = require("./process");
+uglify.consolidator = require("./consolidator");
 
 module.exports = uglify
--- a/bin/uglifyjs
+++ b/bin/uglifyjs
@@ -3,7 +3,7 @@
 
 global.sys = require(/^v0\.[012]/.test(process.version) ? "sys" : "util");
 var fs = require("fs"), path = require("path");
-var uglify = require("../uglify-js"), // symlink ~/.node_libraries/uglify-js.js to ../uglify-js.js
+var uglify = require("uglify-js"),
     consolidator = uglify.consolidator,
     jsp = uglify.parser,
     pro = uglify.uglify;
