https://bugs.launchpad.net/ubuntu/+source/android-headers/+bug/1331435

Index: android-headers/system/graphics.h
===================================================================
--- android-headers.orig/system/graphics.h
+++ android-headers/system/graphics.h
@@ -18,6 +18,8 @@
 #define SYSTEM_CORE_INCLUDE_ANDROID_GRAPHICS_H
 
 #include <stdint.h>
+#include <stddef.h>
+
 
 #ifdef __cplusplus
 extern "C" {
