diff --git a/dart/third_party/boringssl/src/crypto/bio/socket_helper.c b/dart/third_party/boringssl/src/crypto/bio/socket_helper.c
index b1cdd1a..481278f 100644
--- a/dart/third_party/boringssl/src/crypto/bio/socket_helper.c
+++ b/dart/third_party/boringssl/src/crypto/bio/socket_helper.c
@@ -12,7 +12,7 @@
  * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE. */
 
-#define _POSIX_SOURCE
+#define _POSIX_C_SOURCE 200112L
 
 #include <openssl/bio.h>
 #include <openssl/err.h>
