From patchwork Tue Dec  4 21:52:17 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [1/5] meson: remove duplicate definition
From: Dylan Baker <dylan@pnwbakers.com>
X-Patchwork-Id: 265799
Message-Id: <20181204215221.18155-2-dylan@pnwbakers.com>
To: mesa-dev@lists.freedesktop.org
Cc: ignatenko.brain@gmail.com, cefiar@gmail.com
Date: Tue,  4 Dec 2018 13:52:17 -0800

---
 meson.build | 2 --
 1 file changed, 2 deletions(-)

diff --git a/meson.build b/meson.build
index 1aeef95f722..48ba8127f95 100644
--- a/meson.build
+++ b/meson.build
@@ -34,8 +34,6 @@ cpp = meson.get_compiler('cpp')
 
 null_dep = dependency('', required : false)
 
-system_has_kms_drm = ['openbsd', 'netbsd', 'freebsd', 'dragonfly', 'linux'].contains(host_machine.system())
-
 # Arguments for the preprocessor, put these in a separate array from the C and
 # C++ (cpp in meson terminology) arguments since they need to be added to the
 # default arguments for both C and C++.
