diff --git a/examples/rc_test_buttons/Makefile b/examples/rc_test_buttons/Makefile
index 5918b83..90e9e32 100644
--- a/examples/rc_test_buttons/Makefile
+++ b/examples/rc_test_buttons/Makefile
@@ -1,5 +1,5 @@
 # This is a general use makefile for robotics cape projects written in C.
 # Just change the target name to match your main source code filename.
-TARGET = test_buttons
+TARGET = rc_test_buttons
 
 include ../robotics.mk 
