From 14f6c020ccbf1f45b7743773a1931404aa8ea6fb Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 26 Apr 2016 20:48:25 -0500
Subject: [PATCH 01/12] use_pkg-config_for_pcsc-lite_module

---
 wpa_supplicant/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wpa_supplicant/Makefile b/wpa_supplicant/Makefile
index 470a85d..4b26bd8 100644
--- a/wpa_supplicant/Makefile
+++ b/wpa_supplicant/Makefile
@@ -900,7 +900,7 @@ ifdef CONFIG_NATIVE_WINDOWS
 #dynamic symbol loading that is now used in pcsc_funcs.c
 #LIBS += -lwinscard
 else
-LIBS += -lpcsclite -lpthread
+LIBS += $(shell pkg-config --libs libpcsclite)
 endif
 endif
 
-- 
2.8.0.rc3

