From f547a244403aedcc8f8e76e7b5a8e957fb21f237 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 26 Apr 2016 20:52:46 -0500
Subject: [PATCH 12/13] wpasupplicant_band_selection_8b2b718d

---
 wpa_supplicant/scan.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wpa_supplicant/scan.c b/wpa_supplicant/scan.c
index d7e7e90..aa4c489 100644
--- a/wpa_supplicant/scan.c
+++ b/wpa_supplicant/scan.c
@@ -1813,7 +1813,7 @@ static unsigned int max_ht40_rate(int snr)
 		return 81000; /* HT40 MCS4 */
 	if (snr < 22)
 		return 108000; /* HT40 MCS5 */
-	if (snr < 22)
+	if (snr < 24)
 		return 121500; /* HT40 MCS6 */
 	return 135000; /* HT40 MCS7 */
 }
-- 
2.8.0.rc3

