From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Mon, 16 Nov 2015 17:54:20 +0100
Subject: Fix location of completion scripts

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 2205607..1c352ef 100644
--- a/setup.py
+++ b/setup.py
@@ -43,10 +43,10 @@ def _resolve_prefix(prefix, type):
 def get_data_files(prefix):
     data_files = []
     bash_comp_dest = os.path.join(_resolve_prefix(prefix, 'bash_comp'),
-                                  'etc/bash_completion.d')
+                                  'share/bash-completion/completions')
     data_files.append((bash_comp_dest, ['completion/wstool-completion.bash']))
     zsh_comp_dest = os.path.join(_resolve_prefix(prefix, 'zsh_comp'),
-                                 'share/zsh/site-functions')
+                                 'share/zsh/vendor-completions')
     data_files.append((zsh_comp_dest, ['completion/_wstool',
                                        'completion/wstool-completion.bash']))
     return data_files
