From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Mon, 8 Dec 2014 20:02:55 +0100
Subject: Add CMakeLists.txt

---
 CMakeLists.txt | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..6eb85af
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,23 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(ros-comm)
+
+add_subdirectory(tools/rosgraph)
+add_subdirectory(tools/rosconsole)
+add_subdirectory(utilities/xmlrpcpp)
+add_subdirectory(clients/roscpp)
+add_subdirectory(clients/rospy)
+add_subdirectory(tools/rostest)
+add_subdirectory(utilities/message_filters)
+add_subdirectory(utilities/roslz4)
+add_subdirectory(utilities/roswtf)
+add_subdirectory(tools/rosbag_storage)
+add_subdirectory(tools/topic_tools)
+add_subdirectory(tools/rosbag)
+add_subdirectory(tools/roslaunch)
+add_subdirectory(tools/rosmaster)
+add_subdirectory(tools/rosmsg)
+add_subdirectory(tools/rosnode)
+add_subdirectory(tools/rosout)
+add_subdirectory(tools/rosparam)
+add_subdirectory(tools/rosservice)
+add_subdirectory(tools/rostopic)
