Subject: Fix confusing copyright messages/dates
Author: Wookey <wookey@debian.org>
Last-update: Mon, 21 Dec 2015 16:25:55 +0100
Forwarded: No

Should be upstreamed.

Index: ros-ros-comm/test/test_roscpp/test/src/latching_publisher.cpp
===================================================================
--- ros-ros-comm.orig/test/test_roscpp/test/src/latching_publisher.cpp
+++ ros-ros-comm/test/test_roscpp/test/src/latching_publisher.cpp
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2NULLNULL8, Willow Garage, Inc.
+ * Copyright (c) 2008, Willow Garage, Inc.
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
Index: ros-ros-comm/test/test_roscpp/test/src/timer_callbacks.cpp
===================================================================
--- ros-ros-comm.orig/test/test_roscpp/test/src/timer_callbacks.cpp
+++ ros-ros-comm/test/test_roscpp/test/src/timer_callbacks.cpp
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2NULLNULL8, Willow Garage, Inc.
+ * Copyright (c) 2008, Willow Garage, Inc.
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
Index: ros-ros-comm/test/test_roscpp/test/src/wait_for_message.cpp
===================================================================
--- ros-ros-comm.orig/test/test_roscpp/test/src/wait_for_message.cpp
+++ ros-ros-comm/test/test_roscpp/test/src/wait_for_message.cpp
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2NULLNULL8, Willow Garage, Inc.
+ * Copyright (c) 2008, Willow Garage, Inc.
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
