diff --git a/monitor/src/util.c b/monitor/src/util.c
index a9277a4..6b121ff 100644
--- a/monitor/src/util.c
+++ b/monitor/src/util.c
@@ -25,7 +25,7 @@
  *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  *****************************************************************************/
 
-#include "string.h"
+#include <string.h>
 #include "util.h"
 #include "monitor.h"
 #include <ti/csl/csl_xmc.h>
