Description: fix EGL break on Debian X32
Origin: upstream, https://code.qt.io/cgit/qt/qtbase.git/commit/?id=92805a0e9c488e47
Last-Update: 2016-11-15

--- a/src/platformsupport/eglconvenience/qt_egl_p.h
+++ b/src/platformsupport/eglconvenience/qt_egl_p.h
@@ -83,7 +83,7 @@
     { return v; }
 };
 
-#if Q_PROCESSOR_WORDSIZE > 4
+#if QT_POINTER_SIZE > 4
 template <>
 struct QtEglConverter<uintptr_t, uint32_t>
 {
