From 157f7fcf17ce0ab5dfef61bd9ca2afa7746d48d4 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 8 Aug 2018 14:09:37 -0500
Subject: [PATCH 2/2] fix: opencl-firmware git repo

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makefile b/makefile
index eb14ee0..e283382 100644
--- a/makefile
+++ b/makefile
@@ -130,7 +130,7 @@ eve_submodule:
 	git submodule update --init
 
 opencl-firmware:
-	git clone --depth 1 ssh://git@bitbucket.itg.ti.com/mctools/opencl-firmware.git
+	git clone --depth 1 git://git.ti.com/opencl/opencl-firmware.git
 
 change:
 	git log --pretty=format:"- %s%n%b" $(TAG).. ; \
-- 
2.18.0

