Index: skia/BUILD.gn
diff --git a/skia/BUILD.gn b/skia/BUILD.gn
index c6890d2..136071d 100644
--- a/skia/BUILD.gn
+++ b/skia/BUILD.gn
@@ -579,13 +579,6 @@ source_set("skia_opts") {
           "//third_party/skia/src/opts/SkMorphology_opts_neon.cpp",
         ]
 
-        cflags += [
-          # The neon assembly contains conditional instructions which aren't
-          # enclosed in an IT block. The assembler complains without this
-          # option. See #86592.
-          "-Wa,-mimplicit-it=always",
-        ]
-
         # Root build config sets -mfpu=$arm_fpu, which we expect to be neon
         # when running this.
         assert(arm_fpu == "neon")
diff --git a/skia/skia_library_opts.gyp b/skia/skia_library_opts.gyp
index 1c9713f..6dec742 100644
--- a/skia/skia_library_opts.gyp
+++ b/skia/skia_library_opts.gyp
@@ -73,13 +73,6 @@
               ],
             }],
             [ 'arm_version >= 7 and (arm_neon == 1 or arm_neon_optional == 1)', {
-              'cflags': [
-                # The neon assembly contains conditional instructions which
-                # aren't enclosed in an IT block. The assembler complains
-                # without this option.
-                # See #86592.
-                '-Wa,-mimplicit-it=always',
-              ],
               'dependencies': [
                 'skia_opts_neon',
               ]
