From f831f27c309fb2f4dfca3336a9f144993e02c225 Mon Sep 17 00:00:00 2001
From: YASUOKA Masahiko <yasuoka@yasuoka.net>
Date: Wed, 16 Mar 2016 13:39:19 +0900
Subject: [PATCH 10/16] Fix a typo in comment.  Diff from Yuuichi Someya.

---
 pppd/fsm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pppd/fsm.c b/pppd/fsm.c
index c200cc3a8438..e9bd34f0e8f4 100644
--- a/pppd/fsm.c
+++ b/pppd/fsm.c
@@ -468,7 +468,7 @@ fsm_rconfreq(f, id, inp, len)
 	f->nakloops = 0;
 
     } else {
-	/* we sent CONFACK or CONFREJ */
+	/* we sent CONFNAK or CONFREJ */
 	if (f->state != ACKRCVD)
 	    f->state = REQSENT;
 	if( code == CONFNAK )
-- 
2.15.1

