From 2ecc3c7ce01d01e514c8a514b16c45aec1d5613f Mon Sep 17 00:00:00 2001
From: Dmitry Deshevoy <mityada@gmail.com>
Date: Thu, 31 Mar 2016 23:39:32 +0400
Subject: [PATCH 11/16] plog: count only relevant lines from syslog

Closes paulusmack/ppp#42
---
 scripts/plog | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/plog b/scripts/plog
index 84d2c7340cc6..7cb53346413d 100644
--- a/scripts/plog
+++ b/scripts/plog
@@ -3,5 +3,5 @@
 if [ -s /var/log/ppp.log ]; then
   exec tail "$@" /var/log/ppp.log
 else
-  exec tail "$@" /var/log/syslog | grep ' \(pppd\|chat\)\['
+  exec grep ' \(pppd\|chat\)\[' /var/log/syslog | tail "$@"
 fi
-- 
2.15.1

