From 66ae569f1c6fb12df8a463c8beb9afc0c7e5f7f0 Mon Sep 17 00:00:00 2001
From: George Burgess IV <george@gbiv.net>
Date: Fri, 9 Sep 2016 17:36:54 -0700
Subject: [PATCH 14/16] pppd: Remove unused declaration of ttyname.

Signed-off-by: George Burgess IV <george@gbiv.net>
---
 pppd/main.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pppd/main.c b/pppd/main.c
index 6d50d1bac1d9..aef9e8477592 100644
--- a/pppd/main.c
+++ b/pppd/main.c
@@ -257,7 +257,6 @@ static void cleanup_db __P((void));
 static void handle_events __P((void));
 void print_link_stats __P((void));
 
-extern	char	*ttyname __P((int));
 extern	char	*getlogin __P((void));
 int main __P((int, char *[]));
 
-- 
2.15.1

