From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Mon, 16 Nov 2015 23:36:04 +0100
Subject: Add default ROS_ROOT for rosbuild

---
 src/rosinstall/setupfiles.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/rosinstall/setupfiles.py b/src/rosinstall/setupfiles.py
index 6df09cb..c46d9af 100644
--- a/src/rosinstall/setupfiles.py
+++ b/src/rosinstall/setupfiles.py
@@ -290,6 +290,10 @@ EOPYTHON`
 unset _ROS_ROOT_ROSINSTALL
 fi
 
+if [ -z "${ROS_ROOT}" ]; then
+    export ROS_ROOT=/usr/share/ros
+fi
+
 if [ ! -z "$_SETUP_SH_ERROR" ]; then
   # return failure code when sourcing file
   false
