From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Wed, 8 Apr 2015 18:01:46 +0200
Subject: Set default ROS_PACKAGE_PATH to /usr/share

---
 src/rospkg/environment.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rospkg/environment.py b/src/rospkg/environment.py
index 63b7abc..ce340e8 100644
--- a/src/rospkg/environment.py
+++ b/src/rospkg/environment.py
@@ -109,7 +109,7 @@ def get_ros_package_path(env=None):
     """
     if env is None:
         env = os.environ
-    return env.get(ROS_PACKAGE_PATH, None)
+    return env.get(ROS_PACKAGE_PATH, '') + ':/usr/share'
 
 def get_ros_home(env=None):
     """
