From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Wed, 7 Oct 2015 15:29:27 +0200
Subject: Set rosversion to Debian

---
 scripts/rosversion | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/rosversion b/scripts/rosversion
index 78f0677..bfd4a11 100755
--- a/scripts/rosversion
+++ b/scripts/rosversion
@@ -75,7 +75,7 @@ if len(sys.argv) == 2:
         else:
             distro_name = get_distro_name_from_roscore()
         if not distro_name:
-            print('<unknown>')
+            print('Debian')
         else:
             print(distro_name)
         sys.exit(0)
