From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Tue, 24 Nov 2015 13:28:17 +0100
Subject: Fix location of gen_lisp.py

---
 cmake/genlisp-extras.cmake.em | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/genlisp-extras.cmake.em b/cmake/genlisp-extras.cmake.em
index c62037d..b9cdf46 100644
--- a/cmake/genlisp-extras.cmake.em
+++ b/cmake/genlisp-extras.cmake.em
@@ -4,7 +4,7 @@ set(GENLISP_BIN "@(CMAKE_CURRENT_SOURCE_DIR)/scripts/gen_lisp.py")
 set(GENLISP_TEMPLATE_DIR "@(CMAKE_CURRENT_SOURCE_DIR)/scripts")
 @[else]@
 # bin and template dir variables in installspace
-set(GENLISP_BIN "${genlisp_DIR}/../../../@(CATKIN_PACKAGE_BIN_DESTINATION)/gen_lisp.py")
+set(GENLISP_BIN "/usr/lib/genlisp/gen_lisp.py")
 set(GENLISP_TEMPLATE_DIR "${genlisp_DIR}/..")
 @[end if]@
 
