--- a/src/gtk/fm-places-model.c
+++ b/src/gtk/fm-places-model.c
@@ -49,13 +49,13 @@
 /* standard items order */
 typedef enum
 {
+    FM_PLACES_ID_APPLICATIONS,
     FM_PLACES_ID_HOME,
     FM_PLACES_ID_DESKTOP,
-    FM_PLACES_ID_TRASH,
     FM_PLACES_ID_ROOT,
-    FM_PLACES_ID_APPLICATIONS,
     FM_PLACES_ID_COMPUTER,
     FM_PLACES_ID_NETWORK,
+    FM_PLACES_ID_TRASH,
     FM_PLACES_ID_OTHER
 } FmPlacesOrder;
 
