From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Mon, 1 Jun 2015 14:52:06 +0200
Subject: Add CMakeLists.txt

---
 CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..721c5d2
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(bond_core)
+
+add_subdirectory(bond)
+add_subdirectory(smclib)
+add_subdirectory(bondcpp)
+add_subdirectory(bondpy)
+#add_subdirectory(test_bond)
