From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Mon, 1 Jun 2015 14:53:43 +0200
Subject: Add Debian specific SOVERSION

---
 bondcpp/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/bondcpp/CMakeLists.txt b/bondcpp/CMakeLists.txt
index a82ae3d..dd84bd2 100644
--- a/bondcpp/CMakeLists.txt
+++ b/bondcpp/CMakeLists.txt
@@ -21,6 +21,7 @@ add_library(${PROJECT_NAME}
   src/BondSM_sm.cpp
 )
 target_link_libraries(${PROJECT_NAME} ${UUID_LIBRARIES} ${catkin_LIBRARIES} ${BOOST_LIBRARIES})
+set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${bondcpp_VERSION} SOVERSION "0d")
 
 if(catkin_EXPORTED_TARGETS)
   add_dependencies(${PROJECT_NAME} ${catkin_EXPORTED_TARGETS})
