--- src/jdk/src/share/classes/java/lang/Float.java
+++ src/jdk/src/share/classes/java/lang/Float.java
@@ -85,7 +85,9 @@
      *
      * @since 1.6
      */
-    public static final float MIN_NORMAL = 0x1.0p-126f; // 1.17549435E-38f
+    // FIXME: still required on alpha?
+    // public static final float MIN_NORMAL = 0x1.0p-126f; // 1.17549435E-38f
+    public static final float MIN_NORMAL = Float.intBitsToFloat(0x00800000);
 
     /**
      * A constant holding the smallest positive nonzero value of type
