--- openjdk/jdk/src/share/bin/java.c.orig
+++ openjdk/jdk/src/share/bin/java.c
@@ -1580,7 +1580,7 @@
 ReadKnownVMs(const char *jvmCfgName, jboolean speculative)
 {
     FILE *jvmCfg;
-    char line[MAXPATHLEN+20];
+    char line[MAXPATHLEN+30];
     int cnt = 0;
     int lineno = 0;
     jlong start, end;
@@ -1595,6 +1595,10 @@
 
     jvmCfg = fopen(jvmCfgName, "r");
     if (jvmCfg == NULL) {
+      strcat(jvmCfgName, "-default");
+      jvmCfg = fopen(jvmCfgName, "r");
+    }
+    if (jvmCfg == NULL) {
       if (!speculative) {
         JLI_ReportErrorMessage(CFG_ERROR6, jvmCfgName);
         exit(1);
