Index: b/jdk/src/share/bin/java.c
===================================================================
--- a/jdk/src/share/bin/java.c
+++ b/jdk/src/share/bin/java.c
@@ -1698,7 +1698,7 @@ jint
 ReadKnownVMs(const char *jvmCfgName, jboolean speculative)
 {
     FILE *jvmCfg;
-    char line[MAXPATHLEN+20];
+    char line[MAXPATHLEN+30];
     int cnt = 0;
     int lineno = 0;
     jlong start, end;
@@ -1713,6 +1713,10 @@ ReadKnownVMs(const char *jvmCfgName, jbo
 
     jvmCfg = fopen(jvmCfgName, "r");
     if (jvmCfg == NULL) {
+      strcat(jvmCfgName, "-default");
+      jvmCfg = fopen(jvmCfgName, "r");
+    }
+    if (jvmCfg == NULL) {
       if (!speculative) {
         JLI_ReportErrorMessage(CFG_ERROR6, jvmCfgName);
         exit(1);
