# DP: Turn off -Werror for hotspot, overwrites -fpermissive.

Index: b/hotspot/make/linux/makefiles/gcc.make
===================================================================
--- a/hotspot/make/linux/makefiles/gcc.make
+++ b/hotspot/make/linux/makefiles/gcc.make
@@ -197,7 +197,7 @@ else
 endif
 
 # Compiler warnings are treated as errors
-WARNINGS_ARE_ERRORS = -Werror -Wno-error=format -Wno-error=deprecated
+WARNINGS_ARE_ERRORS = 
 
 ifeq ($(USE_CLANG), true)
   # However we need to clean the code up before we can unrestrictedly enable this option with Clang
