Index: b/jdk/src/solaris/classes/sun/awt/X11/XWindowPeer.java
===================================================================
--- a/jdk/src/solaris/classes/sun/awt/X11/XWindowPeer.java
+++ b/jdk/src/solaris/classes/sun/awt/X11/XWindowPeer.java
@@ -1220,6 +1220,7 @@ class XWindowPeer extends XPanelPeer imp
 
     boolean isOverrideRedirect() {
         return XWM.getWMID() == XWM.OPENLOOK_WM ||
+            XWM.getWMID() == XWM.COMPIZ_WM ||
             Window.Type.POPUP.equals(getWindowType());
     }
 
