From: Leopold Palomo-Avellaneda <leopold.palomo@upc.edu>
Date: Wed, 22 Apr 2015 16:04:08 +0200
Subject: Added CMakeLists to build the package

---
 CMakeLists.txt | 13 +++++++++++++
 1 file changed, 13 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..ee46a13
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,13 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(common-msgs)
+
+add_subdirectory(actionlib_msgs)  
+add_subdirectory(common_msgs)
+add_subdirectory(diagnostic_msgs)
+add_subdirectory(geometry_msgs) 
+add_subdirectory(nav_msgs)
+add_subdirectory(sensor_msgs)
+add_subdirectory(shape_msgs) 
+add_subdirectory(stereo_msgs)
+add_subdirectory(trajectory_msgs)
+add_subdirectory(visualization_msgs)
