From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Mon, 23 Nov 2015 14:35:05 +0100
Subject: Use catkin variables for install dirs

---
 CMakeLists.txt | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6f3d12a..5add1dc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,12 +21,10 @@ set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${rosconsole_bridge_VER
 
 install(
   TARGETS ${PROJECT_NAME}
-  ARCHIVE DESTINATION lib
-  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
 )
 
 install(
-  DIRECTORY include/
-  DESTINATION include
-  FILES_MATCHING PATTERN "*.h"
-)
+  DIRECTORY include/${PROJECT_NAME}/
+  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
