diff --git a/dart/runtime/BUILD.gn b/dart/runtime/BUILD.gn
index cbc0643..40e5b17 100644
--- a/dart/runtime/BUILD.gn
+++ b/dart/runtime/BUILD.gn
@@ -32,6 +32,7 @@ config("dart_config") {
     "-Wall",
     "-Wextra", # Also known as -W.
     "-Wno-unused-parameter",
+    "-Wno-unused-variable",
     "-Wnon-virtual-dtor",
     "-Wvla",
     "-Wno-conversion-null",
diff --git a/dart/tools/gyp/configurations.gypi b/dart/tools/gyp/configurations.gypi
index 5b55362..4c49665 100644
--- a/dart/tools/gyp/configurations.gypi
+++ b/dart/tools/gyp/configurations.gypi
@@ -8,6 +8,7 @@
       '-Wall',
       '-Wextra', # Also known as -W.
       '-Wno-unused-parameter',
+      '-Wno-unused-variable',
     ],
 
     # Default value.  This may be overridden in a containing project gyp.
