From: Leopold Palomo-Avellaneda <leopold.palomo@upc.edu>
Date: Tue, 28 Apr 2015 11:42:07 +0200
Subject: Add Debian specific SONAME

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ce896b9..1b27833 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,6 +38,7 @@ catkin_package(LIBRARIES dynamic_reconfigure_config_init_mutex
 add_library(dynamic_reconfigure_config_init_mutex
   src/dynamic_reconfigure_config_init_mutex.cpp)
 target_link_libraries(dynamic_reconfigure_config_init_mutex ${Boost_LIBRARIES})
+set_target_properties(dynamic_reconfigure_config_init_mutex PROPERTIES VERSION ${dynamic_reconfigure_VERSION} SOVERSION "0d")
 
 install(DIRECTORY include/dynamic_reconfigure/
   DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
