Description: Use preferences-system-network as icon
Author: Alf Gaida <agaida@siduction.org>
Bug-Debian: https://bugs.debian.org/751104


--- a/misc/desktop/cmst-autostart.desktop
+++ b/misc/desktop/cmst-autostart.desktop
@@ -5,7 +5,7 @@ Name=Connman UI Setup
 GenericName=Network Configuration
 Comment=QT GUI frontend for connman
 Categories=Settings;System;Qt;Network
-Icon=cmst
+Icon=preferences-system-network
 Exec=cmst
 Terminal=false
 StartupNotify=false
--- a/misc/desktop/cmst.desktop
+++ b/misc/desktop/cmst.desktop
@@ -5,7 +5,7 @@ Name=Connman UI Setup
 GenericName=Network Configuration
 Comment=QT GUI frontend for connman
 Categories=Settings;System;Qt;Network
-Icon=cmst
+Icon=preferences-system-network
 Terminal=false
 Exec=cmst
 StartupNotify=false
