From: Miloslav Trmač <mitr@redhat.com>
Date: Sat, 7 Dec 2013 04:21:10 +0100
Subject: Fix a memory leak

Bug: https://bugs.freedesktop.org/show_bug.cgi?id=72426
Origin: upstream, 0.113, commit:1986e443b170240e9ce4a34726b7fa6c55b3601c
---
 src/polkitbackend/polkitbackendjsauthority.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/polkitbackend/polkitbackendjsauthority.c b/src/polkitbackend/polkitbackendjsauthority.c
index bc2fe22..c3885a9 100644
--- a/src/polkitbackend/polkitbackendjsauthority.c
+++ b/src/polkitbackend/polkitbackendjsauthority.c
@@ -1363,7 +1363,6 @@ js_polkit_spawn (JSContext  *cx,
           goto out;
 	}
       s = JS_EncodeString (cx, JSVAL_TO_STRING (elem_val));
-      s = JS_EncodeString (cx, JSVAL_TO_STRING (elem_val));
       argv[n] = g_strdup (s);
       JS_free (cx, s);
     }
