From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Wed, 8 Apr 2015 18:17:43 +0200
Subject: Remove error when no ROS_PACKAGE_PATH is set

---
 src/rosinstall_generator/generator.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/rosinstall_generator/generator.py b/src/rosinstall_generator/generator.py
index 0ab42d6..99b43b4 100644
--- a/src/rosinstall_generator/generator.py
+++ b/src/rosinstall_generator/generator.py
@@ -189,8 +189,6 @@ _packages_in_environment = None
 def _get_packages_in_environment():
     global _packages_in_environment
     if _packages_in_environment is None:
-        if ROS_PACKAGE_PATH not in os.environ or not os.environ[ROS_PACKAGE_PATH]:
-            raise RuntimeError("The environment variable '%s' must be set when using '%s'" % (ROS_PACKAGE_PATH, ARG_CURRENT_ENVIRONMENT))
         _packages_in_environment = set([])
         rs = RosStack()
         _packages_in_environment.update(set(rs.list()))
