From 8bee9c852f69535bceb3a7fab4dfa9519df7f02c Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 9 Oct 2015 10:43:16 -0500
Subject: [PATCH] systemd: backport 70-power-switch.rules

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 src/login/70-power-switch.rules | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/login/70-power-switch.rules b/src/login/70-power-switch.rules
index 36fb827..e2855b5 100644
--- a/src/login/70-power-switch.rules
+++ b/src/login/70-power-switch.rules
@@ -9,5 +9,10 @@ ACTION=="remove", GOTO="power_switch_end"
 
 SUBSYSTEM=="input", KERNEL=="event*", SUBSYSTEMS=="acpi", TAG+="power-switch"
 SUBSYSTEM=="input", KERNEL=="event*", KERNELS=="thinkpad_acpi", TAG+="power-switch"
+SUBSYSTEM=="input", KERNEL=="event*", ATTRS{name}=="twl4030_pwrbutton", TAG+="power-switch"
+SUBSYSTEM=="input", KERNEL=="event*", ATTRS{name}=="tps65217_pwr_but", TAG+="power-switch"
+SUBSYSTEM=="input", KERNEL=="event*", ATTRS{name}=="* WMI hotkeys", TAG+="power-switch"
+SUBSYSTEM=="input", KERNEL=="event*", \
+  SUBSYSTEMS=="platform", DRIVERS=="gpio-keys", ATTRS{keys}=="*,116|116,*|116|*,116,*", TAG+="power-switch"
 
 LABEL="power_switch_end"
-- 
2.6.1

