From: Jon Severinsson <jon@severinsson.net>
Date: Fri, 11 Jul 2014 02:15:37 +0200
Subject: Map rcS.d init script dependencies to their systemd equivalent.

---
 src/sysv-generator/sysv-generator.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/sysv-generator/sysv-generator.c b/src/sysv-generator/sysv-generator.c
index ea864c1..635e0c7 100644
--- a/src/sysv-generator/sysv-generator.c
+++ b/src/sysv-generator/sysv-generator.c
@@ -283,11 +283,17 @@ static int sysv_translate_facility(const char *name, const char *filename, char
                 "named",                SPECIAL_NSS_LOOKUP_TARGET,
                 "portmap",              SPECIAL_RPCBIND_TARGET,
                 "remote_fs",            SPECIAL_REMOTE_FS_TARGET,
-                "syslog",               NULL,
+                "syslog",               "systemd-journald-dev-log.socket",
                 "time",                 SPECIAL_TIME_SYNC_TARGET,
                 /* Debian defined facilities */
                 "x-display-manager",    "display-manager.service",
                 "mail-transport-agent", "mail-transport-agent.target",
+                /* special rcS init scripts */
+                "cryptdisks",           "cryptsetup.target",
+                "mountall",             SPECIAL_LOCAL_FS_TARGET,
+                "mountnfs",             SPECIAL_REMOTE_FS_TARGET,
+                "checkroot",            "systemd-remount-fs.service",
+                "dbus",                 "dbus.socket",
         };
 
         unsigned i;
