Description: cmd-shim is only needed on win32
Author: Jérémy Lal <kapouer@melix.org>
Forwarded: not-needed, todo: better patch
Last-Update: 2013-09-08
--- a/lib/build.js
+++ b/lib/build.js
@@ -16,8 +16,8 @@
   , readJson = require("read-package-json")
   , link = require("./utils/link.js")
   , linkIfExists = link.ifExists
-  , cmdShim = require("cmd-shim")
-  , cmdShimIfExists = cmdShim.ifExists
+//  , cmdShim = require("cmd-shim")
+//  , cmdShimIfExists = cmdShim.ifExists
   , asyncMap = require("slide").asyncMap
 
 module.exports = build
