From: Leopold Palomo-Avellaneda <leopold.palomo@upc.edu>
Date: Wed, 22 Apr 2015 09:02:30 +0200
Subject: Added CMakeLists to build the package

---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..023ab85
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,5 @@
+CMAKE_MINIMUM_REQUIRED(VERSION 2.6.0)
+
+PROJECT(orocos-kdl)
+add_subdirectory(orocos_kdl)
+
