From: Giuseppe Iuculano <iuculano@debian.org>
Date: Thu, 27 Sep 2018 19:03:07 +0800
Subject: _remove-Id-from-smartd.conf

---
 smartd.conf | 2 --
 1 file changed, 2 deletions(-)

diff --git a/smartd.conf b/smartd.conf
index cd0bfcb..5490bde 100644
--- a/smartd.conf
+++ b/smartd.conf
@@ -2,8 +2,6 @@
 
 # Home page is: http://www.smartmontools.org
 
-# $Id: smartd.conf 4120 2015-08-27 16:12:21Z samm2 $
-
 # smartd will re-read the configuration file if it receives a HUP
 # signal
 
