From: Giuseppe Iuculano <iuculano@debian.org>
Date: Thu, 27 Sep 2018 19:03:07 +0800
Subject: kfreebsd

---
 cciss.cpp | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/cciss.cpp b/cciss.cpp
index f5871dd..62fcba7 100644
--- a/cciss.cpp
+++ b/cciss.cpp
@@ -18,18 +18,10 @@
 #  ifndef be32toh
 #    define be32toh __be32_to_cpu
 #  endif
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 #  include <sys/endian.h>
 #  include CISS_LOCATION
 #  define _HAVE_CCISS
-#elif defined(__FreeBSD_kernel__)
-#  include <endian.h>
-#  ifdef __GLIBC__
-#  include <bsd/sys/cdefs.h>
-#  include <stdint.h>
-#  endif
-#  include CISS_LOCATION
-#  define _HAVE_CCISS
 #endif
 
 #ifdef _HAVE_CCISS
