# Don't build/run PD tests.

Index: a/configure.ac
===================================================================
--- a/configure.ac	2014-05-24 18:32:46.149671334 +0200
+++ b/configure.ac	2014-05-24 18:47:41.446823725 +0200
@@ -139,6 +139,8 @@
 AC_SUBST([LIBSIGROK_CFLAGS])
 AC_SUBST([LIBSIGROK_LIBS])
 
+build_runtc="no"
+
 # Checks for header files.
 # These are already checked: inttypes.h stdint.h stdlib.h string.h unistd.h.
 AC_CHECK_HEADER([sys/resource.h], [], [build_runtc="no"])
