# Don't build/run PD tests.

diff --git a/configure.ac b/configure.ac
index cd76e2a..75b8b14 100644
--- a/configure.ac
+++ b/configure.ac
@@ -134,6 +134,8 @@ PKG_CHECK_MODULES([check], [check >= 0.9.4],
 	[have_check="yes"], [have_check="no"])
 AM_CONDITIONAL(HAVE_CHECK, test x"$have_check" = "xyes")
 
+build_runtc="no"
+
 # Checks for header files.
 # These are already checked: inttypes.h stdint.h stdlib.h string.h unistd.h.
 
