Description: Fix path to nodejs binary on Debian systems
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2014-08-23

Index: node-rimraf/bin.js
===================================================================
--- node-rimraf.orig/bin.js
+++ node-rimraf/bin.js
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
+#!/usr/bin/env nodejs
 
 var rimraf = require('./')
 
Index: node-rimraf/test/run.sh
===================================================================
--- node-rimraf.orig/test/run.sh
+++ node-rimraf/test/run.sh
@@ -4,7 +4,7 @@ code=0
 for i in test-*.js; do
   echo -n $i ...
   bash setup.sh
-  node $i
+  nodejs $i
   if [ -d target ]; then
     echo "fail"
     code=1
