From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Tue, 4 Aug 2015 21:36:50 +0200
Subject: Add Debian specific SOVERSION

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 28ab530..57c518c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,6 +31,7 @@ include_directories(include
 
 add_library(laser_geometry src/laser_geometry.cpp)
 target_link_libraries(laser_geometry ${Boost_LIBRARIES} ${tf_LIBRARIES})
+set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${laser_geometry_VERSION} SOVERSION "0d")
 
 if(CATKIN_ENABLE_TESTING)
   catkin_add_gtest(projection_test test/projection_test.cpp)
