Index: systemtap-2.8/dtrace.in
===================================================================
--- systemtap-2.8.orig/dtrace.in
+++ systemtap-2.8/dtrace.in
@@ -417,7 +417,7 @@ def main():
         cc1 = os.environ.get("CC", "gcc")
         cflags = "-g " + os.environ.get("CFLAGS", "")
         retcode = call(split(cc1) + defines + includes + split(cflags) +
-             ["-fPIC", "-I.", "-I@prefix@/include", "-c", fname, "-o",
+             ["-fPIC", "-I.", "-c", fname, "-o",
               filename + suffix], shell=False)
         if retcode != 0:
             print("\"gcc " + fname + "\" failed")
