From 0e4ef24fba4e9ae5b632d42ab53337104c5d5b6f Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 12 Nov 2015 09:42:00 -0600
Subject: [PATCH] ax_pthread

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 Makefile.am      |   2 +
 configure.ac     |   2 +
 2 files changed, 4 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index 5272bdb..857e15c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -35,6 +35,8 @@
 SUBDIRS =  linux/etc linux/src/utils linux/src/api linux/src/transport \
         linux/src/mm linux/src/daemon linux/src/tests
 
+ACLOCAL_AMFLAGS = -I m4
+
 # where to install common headers on the system
 ipcincludedir = $(includedir)/ti/ipc
 ipcincludeinterfacesdir = $(includedir)/ti/ipc/interfaces
diff --git a/configure.ac b/configure.ac
index e6cb19f..df3b1c4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -43,6 +43,8 @@ AC_CONFIG_SRCDIR([packages/ti/ipc/MessageQ.h])
 # Set auxiliary build tools directory
 AC_CONFIG_AUX_DIR([linux/build-aux])
 
+AC_SEARCH_LIBS([pthread_create], [pthread])
+
 # Initilize automake
 AM_INIT_AUTOMAKE([foreign])
 
-- 
2.6.2

