From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Wed, 10 Dec 2014 08:57:20 +0100
Subject: Search in "/" for message generators

---
 cmake/genmsg-extras.cmake.em | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/genmsg-extras.cmake.em b/cmake/genmsg-extras.cmake.em
index 31fae76..c9871b7 100644
--- a/cmake/genmsg-extras.cmake.em
+++ b/cmake/genmsg-extras.cmake.em
@@ -20,7 +20,7 @@ include(CMakeParseArguments)
 
 # find message generators in all workspaces
 set(message_generators "")
-foreach(workspace ${CATKIN_WORKSPACES})
+foreach(workspace ${CATKIN_WORKSPACES} "/")
   file(GLOB workspace_message_generators
     RELATIVE ${workspace}/${GENMSG_LANGS_DESTINATION}
     ${workspace}/${GENMSG_LANGS_DESTINATION}/gen*)
