Description: kfreebsd port
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2015-09-19
Forwarded: not yet
--- a/configure.ac
+++ b/configure.ac
@@ -53,6 +53,7 @@
 AM_CONDITIONAL([DARWIN],   [AS_CASE([$host_os],[darwin*],       [true], [false])])
 AM_CONDITIONAL([DRAGONFLY],[AS_CASE([$host_os],[dragonfly*],    [true], [false])])
 AM_CONDITIONAL([FREEBSD],  [AS_CASE([$host_os],[freebsd*],      [true], [false])])
+AM_CONDITIONAL([FREEBSD],  [AS_CASE([$host_os],[*freebsd*],      [true], [false])])
 AM_CONDITIONAL([LINUX],    [AS_CASE([$host_os],[linux*],        [true], [false])])
 AM_CONDITIONAL([NETBSD],   [AS_CASE([$host_os],[netbsd*],       [true], [false])])
 AM_CONDITIONAL([OPENBSD],  [AS_CASE([$host_os],[openbsd*],      [true], [false])])
--- a/src/unix/core.c
+++ b/src/unix/core.c
@@ -55,7 +55,7 @@
 # include <sys/ioctl.h>
 #endif
 
-#if defined(__FreeBSD__) || defined(__DragonFly__)
+#if defined(__FreeBSD__) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)
 # include <sys/sysctl.h>
 # include <sys/filio.h>
 # include <sys/ioctl.h>
@@ -500,7 +500,7 @@
 
 
 #if defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__) || \
-    defined(_AIX) || defined(__DragonFly__)
+    defined(_AIX) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)
 
 int uv__nonblock(int fd, int set) {
   int r;
@@ -530,7 +530,7 @@
 }
 
 #else /* !(defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__) || \
-	   defined(_AIX) || defined(__DragonFly__)) */
+	   defined(_AIX) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)) */
 
 int uv__nonblock(int fd, int set) {
   int flags;
@@ -594,7 +594,7 @@
 }
 
 #endif /* defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__) || \
-	  defined(_AIX) || defined(__DragonFly__) */
+	  defined(_AIX) || defined(__DragonFly__) || defined(__FreeBSD_kernel__) */
 
 
 /* This function is not execve-safe, there is a race window
@@ -931,7 +931,7 @@
   int fd;
 
 #if defined(__linux__) || (defined(__FreeBSD__) && __FreeBSD__ >= 9) || \
-    defined(__DragonFly__)
+    defined(__DragonFly__) || defined(__FreeBSD_kernel__)
   static int no_cloexec;
 
   if (!no_cloexec) {
--- a/include/uv-unix.h
+++ b/include/uv-unix.h
@@ -54,7 +54,7 @@
 #elif defined(__APPLE__)
 # include "uv-darwin.h"
 #elif defined(__DragonFly__)  || \
-      defined(__FreeBSD__)    || \
+      defined(__FreeBSD_kernel__)    || \
       defined(__OpenBSD__)    || \
       defined(__NetBSD__)
 # include "uv-bsd.h"
--- a/src/unix/fs.c
+++ b/src/unix/fs.c
@@ -192,6 +192,7 @@
 #elif defined(__APPLE__)                                                      \
     || defined(__DragonFly__)                                                 \
     || defined(__FreeBSD__)                                                   \
+    || defined(__FreeBSD_kernel__)                                            \
     || defined(__NetBSD__)                                                    \
     || defined(__OpenBSD__)                                                   \
     || defined(__sun)
--- a/test/test-fs-event.c
+++ b/test/test-fs-event.c
@@ -29,6 +29,7 @@
 # if defined(__APPLE__) ||                                                    \
      defined(__DragonFly__) ||                                                \
      defined(__FreeBSD__) ||                                                  \
+     defined(__FreeBSD_kernel__) ||                                           \
      defined(__OpenBSD__) ||                                                  \
      defined(__NetBSD__)
 #  define HAVE_KQUEUE 1
